// @sdkwork/ui-pc-react/dist/ui-catalog.js
//#region src/components/ui/catalog.ts
var e = {
	actions: [
		"Button",
		"IconButton",
		"ToolbarButton",
		"SplitButton",
		"ActionMenuButton",
		"BulkActionBar",
		"Command",
		"CommandDialog",
		"CommandEmpty",
		"CommandInput",
		"CommandList",
		"CommandGroup",
		"CommandItem",
		"CommandSeparator",
		"CommandShortcut"
	],
	dataEntry: /* @__PURE__ */ "Input.Textarea.Checkbox.Switch.Select.SelectTrigger.SelectValue.SelectContent.SelectGroup.SelectLabel.SelectItem.SelectSeparator.Label.RadioGroup.RadioGroupItem.Slider.FileUpload.ImageUpload.VideoUpload.AudioUpload.DocumentUpload.Combobox.DateInput.DateInputCalendarButton.DateTimeInput.DateRangeField.DateTimeRangeField.DateRangePicker.NumberInput.SegmentedControl.TagInput.InputGroup.InputGroupAddon".split("."),
	dataDisplay: /* @__PURE__ */ "Badge.Avatar.AvatarImage.AvatarFallback.CollectionGrid.DataTable.RichTree.Timeline.DescriptionList.DescriptionItem.DescriptionTerm.DescriptionDetails.MarkdownViewer.MarkdownViewerEmpty.KeyValueTable.KeyValueTableRow.KeyValueTableLabel.KeyValueTableValue.Tabs.TabsList.TabsTrigger.TabsContent.Table.TableHeader.TableBody.TableFooter.TableRow.TableHead.TableCell.TableCaption.Tree.StatusBadge.StatCard".split("."),
	feedback: [
		"ActivityFeed",
		"ActivityFeedItem",
		"EmptyState",
		"LoadingBlock",
		"StatusNotice",
		"Progress",
		"Skeleton",
		"InlineAlert",
		"EmptySearch",
		"NotificationCenter",
		"NotificationCenterItem",
		"Toaster",
		"SdkworkToaster",
		"toast",
		"sdkToast"
	],
	form: [
		"Form",
		"FormField",
		"FormItem",
		"FormLabel",
		"FormControl",
		"FormDescription",
		"FormMessage",
		"Field",
		"FieldLabel",
		"FieldDescription",
		"FieldError",
		"FieldGroup",
		"FormSection",
		"FormGrid",
		"FormActions",
		"SettingsField",
		"SettingsSection",
		"FilterBar",
		"FilterBarSection",
		"FilterBarActions"
	],
	layout: [
		"Card",
		"CardHeader",
		"CardContent",
		"CardFooter",
		"CardDescription",
		"CardTitle",
		"PanelGroup",
		"Panel",
		"PanelResizeHandle",
		"StatusBar",
		"StatusBarSection",
		"StatusBarItem",
		"Toolbar",
		"ToolbarGroup",
		"ToolbarSpacer",
		"ToolbarSeparator",
		"SidebarSection",
		"ScrollArea",
		"ScrollBar",
		"Separator"
	],
	navigation: /* @__PURE__ */ "Breadcrumb.BreadcrumbList.BreadcrumbItem.BreadcrumbLink.BreadcrumbPage.BreadcrumbSeparator.BreadcrumbEllipsis.Menubar.MenubarCheckboxItem.MenubarGroup.MenubarLabel.MenubarMenu.MenubarPortal.MenubarRadioGroup.MenubarRadioItem.MenubarTrigger.MenubarContent.MenubarItem.MenubarSeparator.MenubarSub.MenubarSubContent.MenubarSubTrigger.MenubarShortcut.Stepper.StepperItem.Pagination.PaginationContent.PaginationItem.PaginationLink.PaginationPrevious.PaginationNext.PaginationEllipsis.WorkspaceTab.WorkspaceTabs".split("."),
	overlays: /* @__PURE__ */ "ContextMenu.ContextMenuTrigger.ContextMenuContent.ContextMenuItem.ContextMenuCheckboxItem.ContextMenuRadioItem.ContextMenuLabel.ContextMenuSeparator.ContextMenuShortcut.ContextMenuGroup.ContextMenuPortal.ContextMenuSub.ContextMenuSubTrigger.ContextMenuSubContent.ContextMenuRadioGroup.Dialog.DialogTrigger.DialogPortal.DialogOverlay.DialogContent.DialogHeader.DialogFooter.DialogTitle.DialogDescription.DialogClose.Modal.ModalTrigger.ModalPortal.ModalOverlay.ModalContent.ModalHeader.ModalBody.ModalFooter.ModalTitle.ModalDescription.ModalClose.ConfirmDialog.Drawer.DrawerTrigger.DrawerPortal.DrawerOverlay.DrawerContent.DrawerHeader.DrawerBody.DrawerFooter.DrawerTitle.DrawerDescription.DrawerClose.HoverCard.HoverCardTrigger.HoverCardContent.Popover.PopoverAnchor.PopoverTrigger.PopoverContent.DropdownMenu.DropdownMenuTrigger.DropdownMenuContent.DropdownMenuItem.DropdownMenuCheckboxItem.DropdownMenuRadioItem.DropdownMenuLabel.DropdownMenuSeparator.DropdownMenuShortcut.DropdownMenuGroup.DropdownMenuPortal.DropdownMenuSub.DropdownMenuSubTrigger.DropdownMenuSubContent.DropdownMenuRadioGroup.Tooltip.TooltipProvider.TooltipTrigger.TooltipContent".split("."),
	patterns: /* @__PURE__ */ "AppShell.PageHeader.NavigationRail.DesktopShellFrame.WorkspacePanel.InspectorPanel.InspectorRail.InspectorRailSection.InspectorRailMetrics.InspectorRailMetric.DesktopWindowControls.DesktopTitleBar.DesktopAppHeader.SettingsCenter.DirtyStateBar.DetailDrawer.OperationDrawer.AnchoredPickerSurface.PickerDialog.EntityPickerDialog.PickerSelectionFooter.TwoPaneSelectorPopover.DetailDrawerSection.DetailDrawerMetrics.DetailDrawerMetric.ListDetailWorkspace.ManagementWorkbench.CrudWorkbench.SearchCommandPalette.RestartRequiredNotice.SectionHeader.WorkspaceScaffold".split("."),
	theme: [
		"SDKWORK_DARK_THEME",
		"SDKWORK_LIGHT_THEME",
		"createSdkworkTheme",
		"createThemeCssVariables",
		"SdkworkShellBridgeProvider",
		"SdkworkThemeProvider",
		"useSdkworkShellBridge"
	],
	utils: ["cn"]
};
//#endregion
export { e as uiComponentCatalog };
