// @sdkwork/drive-app-sdk/generated/server-openapi/src/api/assets.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';

import type { AssetActionRequest, AssetCollection, AssetCollectionItem, AssetCollectionPage, AssetItem, AssetPage, AssetRelation, CreateAssetCollectionItemRequest, CreateAssetCollectionRequest, CreateAssetRelationRequest, CreateAssetRequest, DeleteAssetCollectionItemResponse, DeleteAssetRelationResponse, UpdateAssetRequest } from '../types';


export interface AssetsAssetRelationsDeleteParams {
  tenantId: string;
}

export class AssetsAssetRelationsApi {
  private client: HttpClient;

  constructor(client: HttpClient) {
    this.client = client;
  }


/** Create an asset relation */
  async create(assetId: string, body: CreateAssetRelationRequest): Promise<AssetRelation> {
    return this.client.post<AssetRelation>(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/relations`), body, undefined, undefined, 'application/json');
  }

/** Delete an asset relation */
  async delete(assetId: string, relationId: string, params: AssetsAssetRelationsDeleteParams): Promise<DeleteAssetRelationResponse> {
    const query = buildQueryString([
      { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
    ]);
    return this.client.delete<DeleteAssetRelationResponse>(appendQueryString(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/relations/${serializePathParameter(relationId, { name: 'relationId', style: 'simple', explode: false })}`), query));
  }
}

export interface AssetsAssetCollectionItemsDeleteParams {
  tenantId: string;
}

export class AssetsAssetCollectionItemsApi {
  private client: HttpClient;

  constructor(client: HttpClient) {
    this.client = client;
  }


/** Add an asset to a collection */
  async create(collectionId: string, body: CreateAssetCollectionItemRequest): Promise<AssetCollectionItem> {
    return this.client.post<AssetCollectionItem>(appApiPath(`/assets/collections/${serializePathParameter(collectionId, { name: 'collectionId', style: 'simple', explode: false })}/items`), body, undefined, undefined, 'application/json');
  }

/** Remove an asset from a collection */
  async delete(collectionId: string, itemId: string, params: AssetsAssetCollectionItemsDeleteParams): Promise<DeleteAssetCollectionItemResponse> {
    const query = buildQueryString([
      { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
    ]);
    return this.client.delete<DeleteAssetCollectionItemResponse>(appendQueryString(appApiPath(`/assets/collections/${serializePathParameter(collectionId, { name: 'collectionId', style: 'simple', explode: false })}/items/${serializePathParameter(itemId, { name: 'itemId', style: 'simple', explode: false })}`), query));
  }
}

export interface AssetsAssetCollectionsListParams {
  tenantId: string;
  cursor?: string;
  pageSize?: number;
}

export class AssetsAssetCollectionsApi {
  private client: HttpClient;

  constructor(client: HttpClient) {
    this.client = client;
  }


/** List asset collections */
  async list(params: AssetsAssetCollectionsListParams): Promise<AssetCollectionPage> {
    const query = buildQueryString([
      { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
      { name: 'cursor', value: params.cursor, style: 'form', explode: true, allowReserved: false },
      { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
    ]);
    return this.client.get<AssetCollectionPage>(appendQueryString(appApiPath(`/assets/collections`), query));
  }

/** Create an asset collection */
  async create(body: CreateAssetCollectionRequest): Promise<AssetCollection> {
    return this.client.post<AssetCollection>(appApiPath(`/assets/collections`), body, undefined, undefined, 'application/json');
  }
}

export interface AssetsListParams {
  tenantId: string;
  cursor?: string;
  pageSize?: number;
  kind?: string;
  sourceType?: string;
  q?: string;
}

export interface AssetsGetParams {
  tenantId: string;
}

export class AssetsApi {
  private client: HttpClient;
  public readonly assetCollections: AssetsAssetCollectionsApi;
  public readonly assetCollectionItems: AssetsAssetCollectionItemsApi;
  public readonly assetRelations: AssetsAssetRelationsApi;

  constructor(client: HttpClient) {
    this.client = client;
    this.assetCollections = new AssetsAssetCollectionsApi(client);
    this.assetCollectionItems = new AssetsAssetCollectionItemsApi(client);
    this.assetRelations = new AssetsAssetRelationsApi(client);
  }


/** List global assets */
  async list(params: AssetsListParams): Promise<AssetPage> {
    const query = buildQueryString([
      { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
      { name: 'cursor', value: params.cursor, style: 'form', explode: true, allowReserved: false },
      { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
      { name: 'kind', value: params.kind, style: 'form', explode: true, allowReserved: false },
      { name: 'sourceType', value: params.sourceType, style: 'form', explode: true, allowReserved: false },
      { name: 'q', value: params.q, style: 'form', explode: true, allowReserved: false },
    ]);
    return this.client.get<AssetPage>(appendQueryString(appApiPath(`/assets`), query));
  }

/** Create a global asset metadata record */
  async create(body: CreateAssetRequest): Promise<AssetItem> {
    return this.client.post<AssetItem>(appApiPath(`/assets`), body, undefined, undefined, 'application/json');
  }

/** Get a global asset */
  async get(assetId: string, params: AssetsGetParams): Promise<AssetItem> {
    const query = buildQueryString([
      { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
    ]);
    return this.client.get<AssetItem>(appendQueryString(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}`), query));
  }

/** Update a global asset */
  async update(assetId: string, body: UpdateAssetRequest): Promise<AssetItem> {
    return this.client.patch<AssetItem>(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
  }

/** Archive a global asset */
  async archive(assetId: string, body: AssetActionRequest): Promise<AssetItem> {
    return this.client.post<AssetItem>(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/archive`), body, undefined, undefined, 'application/json');
  }

/** Restore an archived global asset */
  async restore(assetId: string, body: AssetActionRequest): Promise<AssetItem> {
    return this.client.post<AssetItem>(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/restore`), body, undefined, undefined, 'application/json');
  }
}

export function createAssetsApi(client: HttpClient): AssetsApi {
  return new AssetsApi(client);
}

function appendQueryString(path: string, rawQueryString: string): string {
  const query = rawQueryString.replace(/^\?+/, '');
  if (!query) {
    return path;
  }
  return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}

interface PathParameterSpec {
  name: string;
  style: string;
  explode: boolean;
}

function serializePathParameter(value: unknown, spec: PathParameterSpec): string {
  if (value === undefined || value === null) {
    return '';
  }

  const style = spec.style || 'simple';
  if (Array.isArray(value)) {
    return serializePathArray(spec.name, value, style, spec.explode);
  }
  if (typeof value === 'object') {
    return serializePathObject(spec.name, value as Record<string, unknown>, style, spec.explode);
  }
  return pathPrefix(spec.name, style, false) + encodePathValue(serializePathPrimitive(value));
}

function serializePathArray(name: string, values: unknown[], style: string, explode: boolean): string {
  const serialized = values
    .filter((item) => item !== undefined && item !== null)
    .map((item) => encodePathValue(serializePathPrimitive(item)));
  if (serialized.length === 0) {
    return pathPrefix(name, style, false);
  }
  if (style === 'matrix') {
    return explode
      ? serialized.map((item) => `;${name}=${item}`).join('')
      : `;${name}=${serialized.join(',')}`;
  }
  return pathPrefix(name, style, false) + serialized.join(explode ? '.' : ',');
}

function serializePathObject(name: string, value: Record<string, unknown>, style: string, explode: boolean): string {
  const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
  if (entries.length === 0) {
    return pathPrefix(name, style, true);
  }
  if (style === 'matrix') {
    return explode
      ? entries.map(([key, entryValue]) => `;${encodePathValue(key)}=${encodePathValue(serializePathPrimitive(entryValue))}`).join('')
      : `;${name}=${entries.flatMap(([key, entryValue]) => [encodePathValue(key), encodePathValue(serializePathPrimitive(entryValue))]).join(',')}`;
  }
  const serialized = explode
    ? entries.map(([key, entryValue]) => `${encodePathValue(key)}=${encodePathValue(serializePathPrimitive(entryValue))}`).join(style === 'label' ? '.' : ',')
    : entries.flatMap(([key, entryValue]) => [encodePathValue(key), encodePathValue(serializePathPrimitive(entryValue))]).join(',');
  return pathPrefix(name, style, true) + serialized;
}

function pathPrefix(name: string, style: string, _objectValue: boolean): string {
  if (style === 'label') return '.';
  if (style === 'matrix') return `;${name}`;
  return '';
}

function encodePathValue(value: string): string {
  return encodeURIComponent(value);
}

function serializePathPrimitive(value: unknown): string {
  if (value instanceof Date) {
    return value.toISOString();
  }
  if (typeof value === 'object') {
    return JSON.stringify(value);
  }
  return String(value);
}
interface QueryParameterSpec {
  name: string;
  value: unknown;
  style: string;
  explode: boolean;
  allowReserved: boolean;
  contentType?: string;
}

function buildQueryString(parameters: QueryParameterSpec[]): string {
  const pairs: string[] = [];
  for (const parameter of parameters) {
    appendSerializedParameter(pairs, parameter);
  }
  return pairs.join('&');
}

function appendSerializedParameter(pairs: string[], parameter: QueryParameterSpec): void {
  if (parameter.value === undefined || parameter.value === null) {
    return;
  }

  if (parameter.contentType) {
    pairs.push(`${encodeQueryComponent(parameter.name)}=${encodeQueryValue(JSON.stringify(parameter.value), parameter.allowReserved)}`);
    return;
  }

  const style = parameter.style || 'form';
  if (style === 'deepObject') {
    appendDeepObjectParameter(pairs, parameter.name, parameter.value, parameter.allowReserved);
    return;
  }

  if (Array.isArray(parameter.value)) {
    appendArrayParameter(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
    return;
  }

  if (typeof parameter.value === 'object') {
    appendObjectParameter(pairs, parameter.name, parameter.value as Record<string, unknown>, style, parameter.explode, parameter.allowReserved);
    return;
  }

  pairs.push(`${encodeQueryComponent(parameter.name)}=${encodeQueryValue(serializePrimitive(parameter.value), parameter.allowReserved)}`);
}

function appendArrayParameter(
  pairs: string[],
  name: string,
  value: unknown[],
  style: string,
  explode: boolean,
  allowReserved: boolean,
): void {
  const values = value
    .filter((item) => item !== undefined && item !== null)
    .map((item) => serializePrimitive(item));
  if (values.length === 0) {
    return;
  }

  if (style === 'form' && explode) {
    for (const item of values) {
      pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(item, allowReserved)}`);
    }
    return;
  }

  pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(values.join(','), allowReserved)}`);
}

function appendObjectParameter(
  pairs: string[],
  name: string,
  value: Record<string, unknown>,
  style: string,
  explode: boolean,
  allowReserved: boolean,
): void {
  const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
  if (entries.length === 0) {
    return;
  }

  if (style === 'form' && explode) {
    for (const [key, entryValue] of entries) {
      pairs.push(`${encodeQueryComponent(key)}=${encodeQueryValue(serializePrimitive(entryValue), allowReserved)}`);
    }
    return;
  }

  const serialized = entries.flatMap(([key, entryValue]) => [key, serializePrimitive(entryValue)]).join(',');
  pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(serialized, allowReserved)}`);
}

function appendDeepObjectParameter(
  pairs: string[],
  name: string,
  value: unknown,
  allowReserved: boolean,
): void {
  if (!value || typeof value !== 'object' || Array.isArray(value)) {
    pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(serializePrimitive(value), allowReserved)}`);
    return;
  }

  for (const [key, entryValue] of Object.entries(value as Record<string, unknown>)) {
    if (entryValue === undefined || entryValue === null) {
      continue;
    }
    pairs.push(`${encodeQueryComponent(`${name}[${key}]`)}=${encodeQueryValue(serializePrimitive(entryValue), allowReserved)}`);
  }
}

function serializePrimitive(value: unknown): string {
  if (value instanceof Date) {
    return value.toISOString();
  }
  if (typeof value === 'object') {
    return JSON.stringify(value);
  }
  return String(value);
}

function encodeQueryComponent(value: string): string {
  return encodeURIComponent(value);
}

function encodeQueryValue(value: string, allowReserved: boolean): string {
  const encoded = encodeURIComponent(value);
  if (!allowReserved) {
    return encoded;
  }
  return encoded.replace(/%3A/gi, ':')
    .replace(/%2F/gi, '/')
    .replace(/%3F/gi, '?')
    .replace(/%23/gi, '#')
    .replace(/%5B/gi, '[')
    .replace(/%5D/gi, ']')
    .replace(/%40/gi, '@')
    .replace(/%21/gi, '!')
    .replace(/%24/gi, '$')
    .replace(/%26/gi, '&')
    .replace(/%27/gi, "'")
    .replace(/%28/gi, '(')
    .replace(/%29/gi, ')')
    .replace(/%2A/gi, '*')
    .replace(/%2B/gi, '+')
    .replace(/%2C/gi, ',')
    .replace(/%3B/gi, ';')
    .replace(/%3D/gi, '=');
}
