// @sdkwork/ui-pc-react/src/components/ui/data-display/data-table/styles.ts
export const dataTableSurfaceClassName =
  'overflow-hidden rounded-[var(--sdk-radius-field)] border border-[var(--sdk-color-border-default)] bg-[var(--sdk-color-surface-panel)] shadow-[var(--sdk-shadow-sm)]';

export const dataTableFooterClassName =
  'flex flex-col gap-3 border-t border-[var(--sdk-color-border-default)] px-4 py-3 sm:flex-row sm:items-center sm:justify-between';

export const dataTableSummaryClassName =
  'flex min-w-0 flex-1 flex-wrap items-center gap-3 text-sm text-[var(--sdk-color-text-secondary)]';

export const dataTablePaginationLayoutClassName =
  'flex flex-col gap-3 sm:flex-row sm:items-center sm:justify-end';

export const dataTablePageSizeClassName =
  'flex items-center gap-2 text-sm text-[var(--sdk-color-text-secondary)]';

export const dataTablePaginationButtonClassName = 'h-9 rounded-[var(--sdk-radius-field)] px-3';

export const dataTableSortButtonClassName =
  'group -mx-2 inline-flex w-[calc(100%+1rem)] items-center justify-between gap-2 rounded-[var(--sdk-radius-field)] px-2 py-1.5 text-left text-[inherit] outline-none transition-colors hover:bg-[var(--sdk-color-surface-panel-muted)] focus-visible:ring-2 focus-visible:ring-[var(--sdk-color-border-focus)]';

export const dataTableSortIconClassName = 'h-4 w-4 shrink-0 text-[var(--sdk-color-text-muted)]';
