// @sdkwork/drive-app-sdk/composed/operations.ts
// Generated by tools/drive_sdk_generator_runner.mjs from the owner-only Drive OpenAPI input.
// This file is outside generated/server-openapi so SDK ownership metadata does not pollute sdkgen output.

export const sdkMetadata = {
  name: "sdkwork-drive-app-sdk",
  packageName: "sdkwork-drive-app-sdk-generated-typescript",
  sdkOwner: "sdkwork-drive",
  apiAuthority: "sdkwork-drive.app",
  language: "typescript",
  standardProfile: "sdkwork-v3",
  baseUrl: "http://127.0.0.1:18080",
  apiPrefix: "/app/v3/api",
  sdkDependencies: [{"workspace":"sdkwork-appbase-app-sdk","role":"appbase-app-capability","required":true,"dependencyMode":"consumer-sdk","apiPrefix":"/app/v3/api","apiAuthority":"sdkwork-appbase.app","generatedTransportImportPolicy":"forbidden","packageByLanguage":{"typescript":"@sdkwork/appbase-app-sdk","rust":"sdkwork-appbase-app-sdk","java":"com.sdkwork:sdkwork-appbase-app-sdk","python":"sdkwork-appbase-app-sdk","go":"github.com/sdkwork/sdkwork-appbase-app-sdk"}}],
};

export const operations = {
  "archiveEntries.extract": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/archive_entries/extract" },
  "archiveEntries.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/archive_entries" },
  "assetCollectionItems.create": { method: "POST", path: "/app/v3/api/assets/collections/{collectionId}/items" },
  "assetCollectionItems.delete": { method: "DELETE", path: "/app/v3/api/assets/collections/{collectionId}/items/{itemId}" },
  "assetCollections.create": { method: "POST", path: "/app/v3/api/assets/collections" },
  "assetCollections.list": { method: "GET", path: "/app/v3/api/assets/collections" },
  "assetRelations.create": { method: "POST", path: "/app/v3/api/assets/{assetId}/relations" },
  "assetRelations.delete": { method: "DELETE", path: "/app/v3/api/assets/{assetId}/relations/{relationId}" },
  "assets.archive": { method: "POST", path: "/app/v3/api/assets/{assetId}/archive" },
  "assets.create": { method: "POST", path: "/app/v3/api/assets" },
  "assets.get": { method: "GET", path: "/app/v3/api/assets/{assetId}" },
  "assets.list": { method: "GET", path: "/app/v3/api/assets" },
  "assets.restore": { method: "POST", path: "/app/v3/api/assets/{assetId}/restore" },
  "assets.update": { method: "PATCH", path: "/app/v3/api/assets/{assetId}" },
  "changes.list": { method: "GET", path: "/app/v3/api/drive/changes" },
  "changes.startPageToken.get": { method: "GET", path: "/app/v3/api/drive/changes/start_page_token" },
  "changes.watch": { method: "POST", path: "/app/v3/api/drive/changes/watch" },
  "commentReplies.create": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}/replies" },
  "commentReplies.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}/replies/{replyId}" },
  "commentReplies.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}/replies/{replyId}" },
  "commentReplies.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}/replies" },
  "commentReplies.update": { method: "PATCH", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}/replies/{replyId}" },
  "comments.create": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/comments" },
  "comments.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}" },
  "comments.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}" },
  "comments.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/comments" },
  "comments.update": { method: "PATCH", path: "/app/v3/api/drive/nodes/{nodeId}/comments/{commentId}" },
  "downloadPackages.create": { method: "POST", path: "/app/v3/api/drive/download_packages" },
  "downloadPackages.downloadUrls.get": { method: "GET", path: "/app/v3/api/drive/download_packages/{packageId}/download_url" },
  "downloadTokens.resolve": { method: "GET", path: "/app/v3/api/drive/download_tokens/{token}" },
  "downloadUrls.create": { method: "POST", path: "/app/v3/api/drive/download_urls" },
  "favorites.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/favorite" },
  "favorites.list": { method: "GET", path: "/app/v3/api/drive/favorites" },
  "favorites.set": { method: "PUT", path: "/app/v3/api/drive/nodes/{nodeId}/favorite" },
  "nodeLabels.apply": { method: "PUT", path: "/app/v3/api/drive/nodes/{nodeId}/labels/{labelId}" },
  "nodeLabels.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/labels" },
  "nodeLabels.remove": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/labels/{labelId}" },
  "nodeProperties.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/properties/{propertyKey}" },
  "nodeProperties.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/properties" },
  "nodeProperties.set": { method: "PUT", path: "/app/v3/api/drive/nodes/{nodeId}/properties/{propertyKey}" },
  "nodes.capabilities.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/capabilities" },
  "nodes.copy": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/copy" },
  "nodes.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}" },
  "nodes.downloadUrls.create": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/download_url" },
  "nodes.files.create": { method: "POST", path: "/app/v3/api/drive/nodes/files" },
  "nodes.folders.create": { method: "POST", path: "/app/v3/api/drive/nodes/folders" },
  "nodes.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}" },
  "nodes.list": { method: "GET", path: "/app/v3/api/drive/spaces/{spaceId}/nodes" },
  "nodes.move": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/move" },
  "nodes.path.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/path" },
  "nodes.shortcuts.create": { method: "POST", path: "/app/v3/api/drive/nodes/shortcuts" },
  "nodes.update": { method: "PATCH", path: "/app/v3/api/drive/nodes/{nodeId}" },
  "nodes.watch": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/watch" },
  "permissions.create": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/permissions" },
  "permissions.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/permissions/{permissionId}" },
  "permissions.effective.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/permissions/effective" },
  "permissions.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/permissions/{permissionId}" },
  "permissions.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/permissions" },
  "permissions.update": { method: "PATCH", path: "/app/v3/api/drive/nodes/{nodeId}/permissions/{permissionId}" },
  "quotas.summary": { method: "GET", path: "/app/v3/api/drive/quotas/summary" },
  "recent.list": { method: "GET", path: "/app/v3/api/drive/recent" },
  "search.query": { method: "GET", path: "/app/v3/api/drive/search" },
  "sharedWithMe.list": { method: "GET", path: "/app/v3/api/drive/shared_with_me" },
  "shareLinks.create": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/share_links" },
  "shareLinks.get": { method: "GET", path: "/app/v3/api/drive/share_links/{shareLinkId}" },
  "shareLinks.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/share_links" },
  "shareLinks.revoke": { method: "DELETE", path: "/app/v3/api/drive/share_links/{shareLinkId}" },
  "shareLinks.update": { method: "PATCH", path: "/app/v3/api/drive/share_links/{shareLinkId}" },
  "spaces.create": { method: "POST", path: "/app/v3/api/drive/spaces" },
  "spaces.delete": { method: "DELETE", path: "/app/v3/api/drive/spaces/{spaceId}" },
  "spaces.get": { method: "GET", path: "/app/v3/api/drive/spaces/{spaceId}" },
  "spaces.list": { method: "GET", path: "/app/v3/api/drive/spaces" },
  "spaces.update": { method: "PATCH", path: "/app/v3/api/drive/spaces/{spaceId}" },
  "trash.empty": { method: "POST", path: "/app/v3/api/drive/trash/empty" },
  "trash.list": { method: "GET", path: "/app/v3/api/drive/trash" },
  "trash.move": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/trash" },
  "trash.restore": { method: "POST", path: "/app/v3/api/drive/trash/{nodeId}/restore" },
  "uploader.uploads.parts.markUploaded": { method: "PUT", path: "/app/v3/api/drive/uploader/uploads/{uploadItemId}/parts/{partNo}" },
  "uploader.uploads.prepare": { method: "POST", path: "/app/v3/api/drive/uploader/uploads" },
  "uploadSessions.abort": { method: "POST", path: "/app/v3/api/drive/upload_sessions/{uploadSessionId}/abort" },
  "uploadSessions.complete": { method: "POST", path: "/app/v3/api/drive/upload_sessions/{uploadSessionId}/complete" },
  "uploadSessions.create": { method: "POST", path: "/app/v3/api/drive/upload_sessions" },
  "uploadSessions.get": { method: "GET", path: "/app/v3/api/drive/upload_sessions/{uploadSessionId}" },
  "uploadSessions.parts.presign": { method: "PUT", path: "/app/v3/api/drive/upload_sessions/{uploadSessionId}/parts/{partNo}" },
  "versions.delete": { method: "DELETE", path: "/app/v3/api/drive/nodes/{nodeId}/versions/{versionId}" },
  "versions.get": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/versions/{versionId}" },
  "versions.list": { method: "GET", path: "/app/v3/api/drive/nodes/{nodeId}/versions" },
  "versions.restore": { method: "POST", path: "/app/v3/api/drive/nodes/{nodeId}/versions/{versionId}/restore" },
  "watchChannels.get": { method: "GET", path: "/app/v3/api/drive/watch_channels/{channelId}" },
  "watchChannels.list": { method: "GET", path: "/app/v3/api/drive/watch_channels" },
  "watchChannels.stop": { method: "POST", path: "/app/v3/api/drive/watch_channels/{channelId}/stop" },
} as const;
