// @sdkwork/drive-app-sdk/generated/server-openapi/dist/index.cjs
'use strict';

var sdkCommon = require('@sdkwork/sdk-common');

class HttpClient extends sdkCommon.BaseHttpClient {
    constructor(config) {
        super(config);
    }
    getInternalAuthConfig() {
        const self = this;
        self.authConfig = self.authConfig || {};
        return self.authConfig;
    }
    getInternalHeaders() {
        const self = this;
        self.config = self.config || {};
        self.config.headers = self.config.headers || {};
        return self.config.headers;
    }
    buildRequestHeaders(headers, contentType) {
        const mergedHeaders = {
            ...(headers ?? {}),
        };
        if (contentType && contentType.toLowerCase() !== 'multipart/form-data') {
            mergedHeaders['Content-Type'] = contentType;
        }
        return Object.keys(mergedHeaders).length > 0 ? mergedHeaders : undefined;
    }
    buildRequestBody(body, contentType) {
        if (body == null) {
            return body;
        }
        const normalizedContentType = (contentType ?? '').toLowerCase();
        if (normalizedContentType === 'application/x-www-form-urlencoded') {
            return this.encodeFormBody(body);
        }
        if (normalizedContentType === 'multipart/form-data') {
            return this.encodeMultipartBody(body);
        }
        return body;
    }
    encodeMultipartBody(body) {
        if (body instanceof FormData) {
            return body;
        }
        const formData = new FormData();
        if (body instanceof Map) {
            for (const [key, value] of body.entries()) {
                this.appendMultipartValue(formData, String(key), value);
            }
            return formData;
        }
        if (typeof body === 'object') {
            const record = body;
            for (const [key, value] of Object.entries(record)) {
                if (this.isMultipartMetadataField(key)) {
                    continue;
                }
                this.appendMultipartValue(formData, key, value, this.resolveMultipartFileName(record, key));
            }
            return formData;
        }
        this.appendMultipartValue(formData, 'value', body);
        return formData;
    }
    appendMultipartValue(formData, key, value, fileName) {
        if (value == null) {
            return;
        }
        if (Array.isArray(value)) {
            value.forEach((item) => this.appendMultipartValue(formData, key, item, fileName));
            return;
        }
        if (value instanceof Blob) {
            if (fileName) {
                formData.append(key, value, fileName);
                return;
            }
            formData.append(key, value);
            return;
        }
        if (value instanceof Date) {
            formData.append(key, value.toISOString());
            return;
        }
        if (typeof value === 'object') {
            formData.append(key, JSON.stringify(value));
            return;
        }
        formData.append(key, String(value));
    }
    resolveMultipartFileName(record, key) {
        const fieldSpecificName = record[`${key}FileName`];
        if (typeof fieldSpecificName === 'string' && fieldSpecificName.trim()) {
            return fieldSpecificName.trim();
        }
        const genericName = record.fileName;
        if (key === 'file' && typeof genericName === 'string' && genericName.trim()) {
            return genericName.trim();
        }
        return undefined;
    }
    isMultipartMetadataField(key) {
        return key === 'fileName' || key.endsWith('FileName');
    }
    encodeFormBody(body) {
        if (body instanceof URLSearchParams) {
            return body.toString();
        }
        if (typeof body === 'string') {
            return body;
        }
        const params = new URLSearchParams();
        if (body instanceof Map) {
            for (const [key, value] of body.entries()) {
                this.appendFormValue(params, String(key), value);
            }
            return params.toString();
        }
        if (typeof body === 'object') {
            for (const [key, value] of Object.entries(body)) {
                this.appendFormValue(params, key, value);
            }
            return params.toString();
        }
        params.append('value', String(body));
        return params.toString();
    }
    appendFormValue(params, key, value) {
        if (value == null) {
            return;
        }
        if (Array.isArray(value)) {
            value.forEach((item) => this.appendFormValue(params, key, item));
            return;
        }
        if (value instanceof Date) {
            params.append(key, value.toISOString());
            return;
        }
        if (typeof value === 'object') {
            params.append(key, JSON.stringify(value));
            return;
        }
        params.append(key, String(value));
    }
    setApiKey(apiKey) {
        const authConfig = this.getInternalAuthConfig();
        const headers = this.getInternalHeaders();
        authConfig.apiKey = apiKey;
        authConfig.tokenManager?.clearTokens?.();
        if (HttpClient.API_KEY_HEADER === 'Authorization' && HttpClient.API_KEY_USE_BEARER) {
            authConfig.authMode = 'apikey';
            return;
        }
        authConfig.authMode = 'dual-token';
        headers[HttpClient.API_KEY_HEADER] = HttpClient.API_KEY_USE_BEARER
            ? `Bearer ${apiKey}`
            : apiKey;
        if (HttpClient.API_KEY_HEADER.toLowerCase() !== 'authorization') {
            delete headers['Authorization'];
        }
    }
    setAuthToken(token) {
        const headers = this.getInternalHeaders();
        if (HttpClient.API_KEY_HEADER.toLowerCase() !== 'authorization') {
            delete headers[HttpClient.API_KEY_HEADER];
        }
        super.setAuthToken(token);
    }
    setAccessToken(token) {
        const headers = this.getInternalHeaders();
        headers[HttpClient.ACCESS_TOKEN_HEADER] = token;
        super.setAccessToken(token);
    }
    setTokenManager(manager) {
        const baseProto = Object.getPrototypeOf(HttpClient.prototype);
        if (typeof baseProto.setTokenManager === 'function') {
            baseProto.setTokenManager.call(this, manager);
            return;
        }
        this.getInternalAuthConfig().tokenManager = manager;
    }
    applySdkworkAuthHeaders(headers) {
        const authConfig = this.getInternalAuthConfig();
        const tokenManager = authConfig.tokenManager;
        const accessToken = tokenManager?.getAccessToken?.();
        if (!accessToken) {
            return headers;
        }
        return {
            ...(headers ?? {}),
            [HttpClient.ACCESS_TOKEN_HEADER]: accessToken,
        };
    }
    async request(path, options = {}) {
        const execute = this.execute;
        if (typeof execute !== 'function') {
            throw new Error('BaseHttpClient execute method is not available');
        }
        const { body, headers, contentType, method = 'GET', ...rest } = options;
        const requestHeaders = this.applySdkworkAuthHeaders(headers);
        return sdkCommon.withRetry(() => execute.call(this, {
            url: path,
            method,
            ...rest,
            body: this.buildRequestBody(body, contentType),
            headers: this.buildRequestHeaders(requestHeaders, body == null ? undefined : contentType),
        }), { maxRetries: 3 });
    }
    async *streamJson(path, options = {}) {
        const stream = sdkCommon.BaseHttpClient.prototype.stream;
        if (typeof stream !== 'function') {
            throw new Error('BaseHttpClient stream method is not available');
        }
        const { body, headers, contentType, method = 'GET', ...rest } = options;
        const authHeaders = this.applySdkworkAuthHeaders(headers);
        const requestHeaders = this.buildRequestHeaders({ Accept: 'text/event-stream', ...(authHeaders ?? {}) }, body == null ? undefined : contentType);
        for await (const data of stream.call(this, path, {
            method,
            ...rest,
            body: this.buildRequestBody(body, contentType),
            headers: requestHeaders,
        })) {
            if (data === '[DONE]') {
                return;
            }
            if (typeof data !== 'string' || data.trim().length === 0) {
                continue;
            }
            yield JSON.parse(data);
        }
    }
    async get(path, params, headers) {
        return this.request(path, { method: 'GET', params, headers });
    }
    async post(path, body, params, headers, contentType) {
        return this.request(path, { method: 'POST', body, params, headers, contentType });
    }
    async put(path, body, params, headers, contentType) {
        return this.request(path, { method: 'PUT', body, params, headers, contentType });
    }
    async delete(path, params, headers) {
        return this.request(path, { method: 'DELETE', params, headers });
    }
    async patch(path, body, params, headers, contentType) {
        return this.request(path, { method: 'PATCH', body, params, headers, contentType });
    }
}
HttpClient.API_KEY_HEADER = 'Access-Token';
HttpClient.ACCESS_TOKEN_HEADER = 'Access-Token';
HttpClient.API_KEY_USE_BEARER = false;
function createHttpClient(config) {
    return new HttpClient(config);
}

const APP_API_PREFIX = '/app/v3/api';
function appApiPath(path) {
    if (!path) {
        return APP_API_PREFIX;
    }
    if (/^https?:\/\//i.test(path)) {
        return path;
    }
    const normalizedPrefixRaw = (APP_API_PREFIX).trim();
    const normalizedPrefix = normalizedPrefixRaw
        ? `/${normalizedPrefixRaw.replace(/^\/+|\/+$/g, '')}`
        : '';
    const normalizedPath = path.startsWith('/') ? path : `/${path}`;
    if (!normalizedPrefix || normalizedPrefix === '/') {
        return normalizedPath;
    }
    if (normalizedPath === normalizedPrefix || normalizedPath.startsWith(`${normalizedPrefix}/`)) {
        return normalizedPath;
    }
    return `${normalizedPrefix}${normalizedPath}`;
}

class DriveUploaderUploadsPartsApi {
    constructor(client) {
        this.client = client;
    }
    async markUploaded(uploadItemId, partNo, body) {
        return this.client.put(appApiPath(`/drive/uploader/uploads/${serializePathParameter$4(uploadItemId, { name: 'uploadItemId', style: 'simple', explode: false })}/parts/${serializePathParameter$4(partNo, { name: 'partNo', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
}
class DriveUploaderUploadsApi {
    constructor(client) {
        this.client = client;
        this.parts = new DriveUploaderUploadsPartsApi(client);
    }
    async prepare(body) {
        return this.client.post(appApiPath(`/drive/uploader/uploads`), body, undefined, undefined, 'application/json');
    }
}
class DriveUploaderApi {
    constructor(client) {
        this.client = client;
        this.uploads = new DriveUploaderUploadsApi(client);
    }
}
class DriveArchiveEntriesApi {
    constructor(client) {
        this.client = client;
    }
    async list(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/archive_entries`), query));
    }
    async extract(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/archive_entries/extract`), body, undefined, undefined, 'application/json');
    }
}
class DriveDownloadPackagesDownloadUrlsApi {
    constructor(client) {
        this.client = client;
    }
    async get(packageId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/download_packages/${serializePathParameter$4(packageId, { name: 'packageId', style: 'simple', explode: false })}/download_url`), query));
    }
}
class DriveDownloadPackagesApi {
    constructor(client) {
        this.client = client;
        this.downloadUrls = new DriveDownloadPackagesDownloadUrlsApi(client);
    }
    async create(body) {
        return this.client.post(appApiPath(`/drive/download_packages`), body, undefined, undefined, 'application/json');
    }
}
class DriveUploadSessionsPartsApi {
    constructor(client) {
        this.client = client;
    }
    async presign(uploadSessionId, partNo, body) {
        return this.client.put(appApiPath(`/drive/upload_sessions/${serializePathParameter$4(uploadSessionId, { name: 'uploadSessionId', style: 'simple', explode: false })}/parts/${serializePathParameter$4(partNo, { name: 'partNo', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
}
class DriveUploadSessionsApi {
    constructor(client) {
        this.client = client;
        this.parts = new DriveUploadSessionsPartsApi(client);
    }
    async create(body) {
        return this.client.post(appApiPath(`/drive/upload_sessions`), body, undefined, undefined, 'application/json');
    }
    async get(uploadSessionId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/upload_sessions/${serializePathParameter$4(uploadSessionId, { name: 'uploadSessionId', style: 'simple', explode: false })}`), query));
    }
    async abort(uploadSessionId, body) {
        return this.client.post(appApiPath(`/drive/upload_sessions/${serializePathParameter$4(uploadSessionId, { name: 'uploadSessionId', style: 'simple', explode: false })}/abort`), body, undefined, undefined, 'application/json');
    }
    async complete(uploadSessionId, body) {
        return this.client.post(appApiPath(`/drive/upload_sessions/${serializePathParameter$4(uploadSessionId, { name: 'uploadSessionId', style: 'simple', explode: false })}/complete`), body, undefined, undefined, 'application/json');
    }
}
class DriveSpacesApi {
    constructor(client) {
        this.client = client;
    }
    async list(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'ownerSubjectType', value: params.ownerSubjectType, style: 'form', explode: true, allowReserved: false },
            { name: 'ownerSubjectId', value: params.ownerSubjectId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/spaces`), query));
    }
    async create(body) {
        return this.client.post(appApiPath(`/drive/spaces`), body, undefined, undefined, 'application/json');
    }
    async get(spaceId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/spaces/${serializePathParameter$4(spaceId, { name: 'spaceId', style: 'simple', explode: false })}`), query));
    }
    async update(spaceId, body) {
        return this.client.patch(appApiPath(`/drive/spaces/${serializePathParameter$4(spaceId, { name: 'spaceId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    async delete(spaceId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/spaces/${serializePathParameter$4(spaceId, { name: 'spaceId', style: 'simple', explode: false })}`), query));
    }
}
class DriveSharedWithMeApi {
    constructor(client) {
        this.client = client;
    }
    async list(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectType', value: params.subjectType, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectId', value: params.subjectId, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/shared_with_me`), query));
    }
}
class DriveSearchApi {
    constructor(client) {
        this.client = client;
    }
    async query(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'q', value: params.q, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/search`), query));
    }
}
class DriveRecentApi {
    constructor(client) {
        this.client = client;
    }
    async list(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/recent`), query));
    }
}
class DriveVersionsApi {
    constructor(client) {
        this.client = client;
    }
    async list(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/versions`), query));
    }
    async delete(nodeId, versionId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/versions/${serializePathParameter$4(versionId, { name: 'versionId', style: 'simple', explode: false })}`), query));
    }
    async get(nodeId, versionId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/versions/${serializePathParameter$4(versionId, { name: 'versionId', style: 'simple', explode: false })}`), query));
    }
    async restore(nodeId, versionId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/versions/${serializePathParameter$4(versionId, { name: 'versionId', style: 'simple', explode: false })}/restore`), body, undefined, undefined, 'application/json');
    }
}
class DriveTrashApi {
    constructor(client) {
        this.client = client;
    }
    async move(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/trash`), body, undefined, undefined, 'application/json');
    }
    async list(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/trash`), query));
    }
    async restore(nodeId, body) {
        return this.client.post(appApiPath(`/drive/trash/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/restore`), body, undefined, undefined, 'application/json');
    }
    async empty(body) {
        return this.client.post(appApiPath(`/drive/trash/empty`), body, undefined, undefined, 'application/json');
    }
}
class DriveShareLinksApi {
    constructor(client) {
        this.client = client;
    }
    async create(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/share_links`), body, undefined, undefined, 'application/json');
    }
    async list(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/share_links`), query));
    }
    async revoke(shareLinkId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/share_links/${serializePathParameter$4(shareLinkId, { name: 'shareLinkId', style: 'simple', explode: false })}`), query));
    }
    async update(shareLinkId, body) {
        return this.client.patch(appApiPath(`/drive/share_links/${serializePathParameter$4(shareLinkId, { name: 'shareLinkId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    async get(shareLinkId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/share_links/${serializePathParameter$4(shareLinkId, { name: 'shareLinkId', style: 'simple', explode: false })}`), query));
    }
}
class DrivePermissionsEffectiveApi {
    constructor(client) {
        this.client = client;
    }
    async list(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/permissions/effective`), query));
    }
}
class DrivePermissionsApi {
    constructor(client) {
        this.client = client;
        this.effective = new DrivePermissionsEffectiveApi(client);
    }
    async list(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/permissions`), query));
    }
    async create(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/permissions`), body, undefined, undefined, 'application/json');
    }
    async delete(nodeId, permissionId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/permissions/${serializePathParameter$4(permissionId, { name: 'permissionId', style: 'simple', explode: false })}`), query));
    }
    async update(nodeId, permissionId, body) {
        return this.client.patch(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/permissions/${serializePathParameter$4(permissionId, { name: 'permissionId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    async get(nodeId, permissionId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/permissions/${serializePathParameter$4(permissionId, { name: 'permissionId', style: 'simple', explode: false })}`), query));
    }
}
class DriveCommentRepliesApi {
    constructor(client) {
        this.client = client;
    }
    async list(nodeId, commentId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}/replies`), query));
    }
    async create(nodeId, commentId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}/replies`), body, undefined, undefined, 'application/json');
    }
    async get(nodeId, commentId, replyId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}/replies/${serializePathParameter$4(replyId, { name: 'replyId', style: 'simple', explode: false })}`), query));
    }
    async update(nodeId, commentId, replyId, body) {
        return this.client.patch(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}/replies/${serializePathParameter$4(replyId, { name: 'replyId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    async delete(nodeId, commentId, replyId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}/replies/${serializePathParameter$4(replyId, { name: 'replyId', style: 'simple', explode: false })}`), query));
    }
}
class DriveCommentsApi {
    constructor(client) {
        this.client = client;
    }
    async list(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments`), query));
    }
    async create(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments`), body, undefined, undefined, 'application/json');
    }
    async get(nodeId, commentId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}`), query));
    }
    async update(nodeId, commentId, body) {
        return this.client.patch(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    async delete(nodeId, commentId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/comments/${serializePathParameter$4(commentId, { name: 'commentId', style: 'simple', explode: false })}`), query));
    }
}
class DriveNodesFoldersApi {
    constructor(client) {
        this.client = client;
    }
    async create(body) {
        return this.client.post(appApiPath(`/drive/nodes/folders`), body, undefined, undefined, 'application/json');
    }
}
class DriveNodesFilesApi {
    constructor(client) {
        this.client = client;
    }
    async create(body) {
        return this.client.post(appApiPath(`/drive/nodes/files`), body, undefined, undefined, 'application/json');
    }
}
class DriveNodesPathApi {
    constructor(client) {
        this.client = client;
    }
    async get(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/path`), query));
    }
}
class DriveNodesDownloadUrlsApi {
    constructor(client) {
        this.client = client;
    }
    async create(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'requestedTtlSeconds', value: params.requestedTtlSeconds, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/download_url`), query));
    }
}
class DriveNodesCapabilitiesApi {
    constructor(client) {
        this.client = client;
    }
    async get(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectType', value: params.subjectType, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectId', value: params.subjectId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/capabilities`), query));
    }
}
class DriveNodesApi {
    constructor(client) {
        this.client = client;
        this.capabilities = new DriveNodesCapabilitiesApi(client);
        this.downloadUrls = new DriveNodesDownloadUrlsApi(client);
        this.path = new DriveNodesPathApi(client);
        this.files = new DriveNodesFilesApi(client);
        this.folders = new DriveNodesFoldersApi(client);
    }
    async update(nodeId, body) {
        return this.client.patch(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    async get(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}`), query));
    }
    async delete(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}`), query));
    }
    async copy(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/copy`), body, undefined, undefined, 'application/json');
    }
    async move(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/move`), body, undefined, undefined, 'application/json');
    }
    async list(spaceId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'parentNodeId', value: params.parentNodeId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/spaces/${serializePathParameter$4(spaceId, { name: 'spaceId', style: 'simple', explode: false })}/nodes`), query));
    }
}
class DriveQuotasApi {
    constructor(client) {
        this.client = client;
    }
    async summary(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/quotas/summary`), query));
    }
}
class DriveFavoritesApi {
    constructor(client) {
        this.client = client;
    }
    async list(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectType', value: params.subjectType, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectId', value: params.subjectId, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/favorites`), query));
    }
    async set(nodeId, body) {
        return this.client.put(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/favorite`), body, undefined, undefined, 'application/json');
    }
    async delete(nodeId, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectType', value: params.subjectType, style: 'form', explode: true, allowReserved: false },
            { name: 'subjectId', value: params.subjectId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$4(appApiPath(`/drive/nodes/${serializePathParameter$4(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/favorite`), query));
    }
}
class DriveDownloadUrlsApi {
    constructor(client) {
        this.client = client;
    }
    async create(body) {
        return this.client.post(appApiPath(`/drive/download_urls`), body, undefined, undefined, 'application/json');
    }
}
class DriveDownloadTokensApi {
    constructor(client) {
        this.client = client;
    }
    async resolve(token, params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/download_tokens/${serializePathParameter$4(token, { name: 'token', style: 'simple', explode: false })}`), query));
    }
}
class DriveChangesStartPageTokenApi {
    constructor(client) {
        this.client = client;
    }
    async get(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/changes/start_page_token`), query));
    }
}
class DriveChangesApi {
    constructor(client) {
        this.client = client;
        this.startPageToken = new DriveChangesStartPageTokenApi(client);
    }
    async list(params) {
        const query = buildQueryString$4([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'spaceId', value: params.spaceId, style: 'form', explode: true, allowReserved: false },
            { name: 'cursor', value: params.cursor, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$4(appApiPath(`/drive/changes`), query));
    }
}
class DriveApi {
    constructor(client) {
        this.client = client;
        this.changes = new DriveChangesApi(client);
        this.downloadTokens = new DriveDownloadTokensApi(client);
        this.downloadUrls = new DriveDownloadUrlsApi(client);
        this.favorites = new DriveFavoritesApi(client);
        this.quotas = new DriveQuotasApi(client);
        this.nodes = new DriveNodesApi(client);
        this.comments = new DriveCommentsApi(client);
        this.commentReplies = new DriveCommentRepliesApi(client);
        this.permissions = new DrivePermissionsApi(client);
        this.shareLinks = new DriveShareLinksApi(client);
        this.trash = new DriveTrashApi(client);
        this.versions = new DriveVersionsApi(client);
        this.recent = new DriveRecentApi(client);
        this.search = new DriveSearchApi(client);
        this.sharedWithMe = new DriveSharedWithMeApi(client);
        this.spaces = new DriveSpacesApi(client);
        this.uploadSessions = new DriveUploadSessionsApi(client);
        this.downloadPackages = new DriveDownloadPackagesApi(client);
        this.archiveEntries = new DriveArchiveEntriesApi(client);
        this.uploader = new DriveUploaderApi(client);
    }
}
function createDriveApi(client) {
    return new DriveApi(client);
}
function appendQueryString$4(path, rawQueryString) {
    const query = rawQueryString.replace(/^\?+/, '');
    if (!query) {
        return path;
    }
    return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}
function serializePathParameter$4(value, spec) {
    if (value === undefined || value === null) {
        return '';
    }
    const style = spec.style || 'simple';
    if (Array.isArray(value)) {
        return serializePathArray$4(spec.name, value, style, spec.explode);
    }
    if (typeof value === 'object') {
        return serializePathObject$4(spec.name, value, style, spec.explode);
    }
    return pathPrefix$4(spec.name, style) + encodePathValue$4(serializePathPrimitive$4(value));
}
function serializePathArray$4(name, values, style, explode) {
    const serialized = values
        .filter((item) => item !== undefined && item !== null)
        .map((item) => encodePathValue$4(serializePathPrimitive$4(item)));
    if (serialized.length === 0) {
        return pathPrefix$4(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? serialized.map((item) => `;${name}=${item}`).join('')
            : `;${name}=${serialized.join(',')}`;
    }
    return pathPrefix$4(name, style) + serialized.join(explode ? '.' : ',');
}
function serializePathObject$4(name, value, style, explode) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return pathPrefix$4(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? entries.map(([key, entryValue]) => `;${encodePathValue$4(key)}=${encodePathValue$4(serializePathPrimitive$4(entryValue))}`).join('')
            : `;${name}=${entries.flatMap(([key, entryValue]) => [encodePathValue$4(key), encodePathValue$4(serializePathPrimitive$4(entryValue))]).join(',')}`;
    }
    const serialized = explode
        ? entries.map(([key, entryValue]) => `${encodePathValue$4(key)}=${encodePathValue$4(serializePathPrimitive$4(entryValue))}`).join(style === 'label' ? '.' : ',')
        : entries.flatMap(([key, entryValue]) => [encodePathValue$4(key), encodePathValue$4(serializePathPrimitive$4(entryValue))]).join(',');
    return pathPrefix$4(name, style) + serialized;
}
function pathPrefix$4(name, style, _objectValue) {
    if (style === 'label')
        return '.';
    if (style === 'matrix')
        return `;${name}`;
    return '';
}
function encodePathValue$4(value) {
    return encodeURIComponent(value);
}
function serializePathPrimitive$4(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function buildQueryString$4(parameters) {
    const pairs = [];
    for (const parameter of parameters) {
        appendSerializedParameter$4(pairs, parameter);
    }
    return pairs.join('&');
}
function appendSerializedParameter$4(pairs, parameter) {
    if (parameter.value === undefined || parameter.value === null) {
        return;
    }
    if (parameter.contentType) {
        pairs.push(`${encodeQueryComponent$4(parameter.name)}=${encodeQueryValue$4(JSON.stringify(parameter.value), parameter.allowReserved)}`);
        return;
    }
    const style = parameter.style || 'form';
    if (style === 'deepObject') {
        appendDeepObjectParameter$4(pairs, parameter.name, parameter.value, parameter.allowReserved);
        return;
    }
    if (Array.isArray(parameter.value)) {
        appendArrayParameter$4(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    if (typeof parameter.value === 'object') {
        appendObjectParameter$4(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    pairs.push(`${encodeQueryComponent$4(parameter.name)}=${encodeQueryValue$4(serializePrimitive$4(parameter.value), parameter.allowReserved)}`);
}
function appendArrayParameter$4(pairs, name, value, style, explode, allowReserved) {
    const values = value
        .filter((item) => item !== undefined && item !== null)
        .map((item) => serializePrimitive$4(item));
    if (values.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const item of values) {
            pairs.push(`${encodeQueryComponent$4(name)}=${encodeQueryValue$4(item, allowReserved)}`);
        }
        return;
    }
    pairs.push(`${encodeQueryComponent$4(name)}=${encodeQueryValue$4(values.join(','), allowReserved)}`);
}
function appendObjectParameter$4(pairs, name, value, style, explode, allowReserved) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const [key, entryValue] of entries) {
            pairs.push(`${encodeQueryComponent$4(key)}=${encodeQueryValue$4(serializePrimitive$4(entryValue), allowReserved)}`);
        }
        return;
    }
    const serialized = entries.flatMap(([key, entryValue]) => [key, serializePrimitive$4(entryValue)]).join(',');
    pairs.push(`${encodeQueryComponent$4(name)}=${encodeQueryValue$4(serialized, allowReserved)}`);
}
function appendDeepObjectParameter$4(pairs, name, value, allowReserved) {
    if (!value || typeof value !== 'object' || Array.isArray(value)) {
        pairs.push(`${encodeQueryComponent$4(name)}=${encodeQueryValue$4(serializePrimitive$4(value), allowReserved)}`);
        return;
    }
    for (const [key, entryValue] of Object.entries(value)) {
        if (entryValue === undefined || entryValue === null) {
            continue;
        }
        pairs.push(`${encodeQueryComponent$4(`${name}[${key}]`)}=${encodeQueryValue$4(serializePrimitive$4(entryValue), allowReserved)}`);
    }
}
function serializePrimitive$4(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function encodeQueryComponent$4(value) {
    return encodeURIComponent(value);
}
function encodeQueryValue$4(value, allowReserved) {
    const encoded = encodeURIComponent(value);
    if (!allowReserved) {
        return encoded;
    }
    return encoded.replace(/%3A/gi, ':')
        .replace(/%2F/gi, '/')
        .replace(/%3F/gi, '?')
        .replace(/%23/gi, '#')
        .replace(/%5B/gi, '[')
        .replace(/%5D/gi, ']')
        .replace(/%40/gi, '@')
        .replace(/%21/gi, '!')
        .replace(/%24/gi, '$')
        .replace(/%26/gi, '&')
        .replace(/%27/gi, "'")
        .replace(/%28/gi, '(')
        .replace(/%29/gi, ')')
        .replace(/%2A/gi, '*')
        .replace(/%2B/gi, '+')
        .replace(/%2C/gi, ',')
        .replace(/%3B/gi, ';')
        .replace(/%3D/gi, '=');
}

class NodeLabelsApi {
    constructor(client) {
        this.client = client;
    }
    /** List labels applied to a node */
    async list(nodeId, params) {
        const query = buildQueryString$3([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'labelKey', value: params.labelKey, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$3(appApiPath(`/drive/nodes/${serializePathParameter$3(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/labels`), query));
    }
    /** Apply a label to a node */
    async apply(nodeId, labelId, body) {
        return this.client.put(appApiPath(`/drive/nodes/${serializePathParameter$3(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/labels/${serializePathParameter$3(labelId, { name: 'labelId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    /** Remove a label from a node */
    async remove(nodeId, labelId, params) {
        const query = buildQueryString$3([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$3(appApiPath(`/drive/nodes/${serializePathParameter$3(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/labels/${serializePathParameter$3(labelId, { name: 'labelId', style: 'simple', explode: false })}`), query));
    }
}
function createNodeLabelsApi(client) {
    return new NodeLabelsApi(client);
}
function appendQueryString$3(path, rawQueryString) {
    const query = rawQueryString.replace(/^\?+/, '');
    if (!query) {
        return path;
    }
    return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}
function serializePathParameter$3(value, spec) {
    if (value === undefined || value === null) {
        return '';
    }
    const style = spec.style || 'simple';
    if (Array.isArray(value)) {
        return serializePathArray$3(spec.name, value, style, spec.explode);
    }
    if (typeof value === 'object') {
        return serializePathObject$3(spec.name, value, style, spec.explode);
    }
    return pathPrefix$3(spec.name, style) + encodePathValue$3(serializePathPrimitive$3(value));
}
function serializePathArray$3(name, values, style, explode) {
    const serialized = values
        .filter((item) => item !== undefined && item !== null)
        .map((item) => encodePathValue$3(serializePathPrimitive$3(item)));
    if (serialized.length === 0) {
        return pathPrefix$3(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? serialized.map((item) => `;${name}=${item}`).join('')
            : `;${name}=${serialized.join(',')}`;
    }
    return pathPrefix$3(name, style) + serialized.join(explode ? '.' : ',');
}
function serializePathObject$3(name, value, style, explode) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return pathPrefix$3(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? entries.map(([key, entryValue]) => `;${encodePathValue$3(key)}=${encodePathValue$3(serializePathPrimitive$3(entryValue))}`).join('')
            : `;${name}=${entries.flatMap(([key, entryValue]) => [encodePathValue$3(key), encodePathValue$3(serializePathPrimitive$3(entryValue))]).join(',')}`;
    }
    const serialized = explode
        ? entries.map(([key, entryValue]) => `${encodePathValue$3(key)}=${encodePathValue$3(serializePathPrimitive$3(entryValue))}`).join(style === 'label' ? '.' : ',')
        : entries.flatMap(([key, entryValue]) => [encodePathValue$3(key), encodePathValue$3(serializePathPrimitive$3(entryValue))]).join(',');
    return pathPrefix$3(name, style) + serialized;
}
function pathPrefix$3(name, style, _objectValue) {
    if (style === 'label')
        return '.';
    if (style === 'matrix')
        return `;${name}`;
    return '';
}
function encodePathValue$3(value) {
    return encodeURIComponent(value);
}
function serializePathPrimitive$3(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function buildQueryString$3(parameters) {
    const pairs = [];
    for (const parameter of parameters) {
        appendSerializedParameter$3(pairs, parameter);
    }
    return pairs.join('&');
}
function appendSerializedParameter$3(pairs, parameter) {
    if (parameter.value === undefined || parameter.value === null) {
        return;
    }
    if (parameter.contentType) {
        pairs.push(`${encodeQueryComponent$3(parameter.name)}=${encodeQueryValue$3(JSON.stringify(parameter.value), parameter.allowReserved)}`);
        return;
    }
    const style = parameter.style || 'form';
    if (style === 'deepObject') {
        appendDeepObjectParameter$3(pairs, parameter.name, parameter.value, parameter.allowReserved);
        return;
    }
    if (Array.isArray(parameter.value)) {
        appendArrayParameter$3(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    if (typeof parameter.value === 'object') {
        appendObjectParameter$3(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    pairs.push(`${encodeQueryComponent$3(parameter.name)}=${encodeQueryValue$3(serializePrimitive$3(parameter.value), parameter.allowReserved)}`);
}
function appendArrayParameter$3(pairs, name, value, style, explode, allowReserved) {
    const values = value
        .filter((item) => item !== undefined && item !== null)
        .map((item) => serializePrimitive$3(item));
    if (values.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const item of values) {
            pairs.push(`${encodeQueryComponent$3(name)}=${encodeQueryValue$3(item, allowReserved)}`);
        }
        return;
    }
    pairs.push(`${encodeQueryComponent$3(name)}=${encodeQueryValue$3(values.join(','), allowReserved)}`);
}
function appendObjectParameter$3(pairs, name, value, style, explode, allowReserved) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const [key, entryValue] of entries) {
            pairs.push(`${encodeQueryComponent$3(key)}=${encodeQueryValue$3(serializePrimitive$3(entryValue), allowReserved)}`);
        }
        return;
    }
    const serialized = entries.flatMap(([key, entryValue]) => [key, serializePrimitive$3(entryValue)]).join(',');
    pairs.push(`${encodeQueryComponent$3(name)}=${encodeQueryValue$3(serialized, allowReserved)}`);
}
function appendDeepObjectParameter$3(pairs, name, value, allowReserved) {
    if (!value || typeof value !== 'object' || Array.isArray(value)) {
        pairs.push(`${encodeQueryComponent$3(name)}=${encodeQueryValue$3(serializePrimitive$3(value), allowReserved)}`);
        return;
    }
    for (const [key, entryValue] of Object.entries(value)) {
        if (entryValue === undefined || entryValue === null) {
            continue;
        }
        pairs.push(`${encodeQueryComponent$3(`${name}[${key}]`)}=${encodeQueryValue$3(serializePrimitive$3(entryValue), allowReserved)}`);
    }
}
function serializePrimitive$3(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function encodeQueryComponent$3(value) {
    return encodeURIComponent(value);
}
function encodeQueryValue$3(value, allowReserved) {
    const encoded = encodeURIComponent(value);
    if (!allowReserved) {
        return encoded;
    }
    return encoded.replace(/%3A/gi, ':')
        .replace(/%2F/gi, '/')
        .replace(/%3F/gi, '?')
        .replace(/%23/gi, '#')
        .replace(/%5B/gi, '[')
        .replace(/%5D/gi, ']')
        .replace(/%40/gi, '@')
        .replace(/%21/gi, '!')
        .replace(/%24/gi, '$')
        .replace(/%26/gi, '&')
        .replace(/%27/gi, "'")
        .replace(/%28/gi, '(')
        .replace(/%29/gi, ')')
        .replace(/%2A/gi, '*')
        .replace(/%2B/gi, '+')
        .replace(/%2C/gi, ',')
        .replace(/%3B/gi, ';')
        .replace(/%3D/gi, '=');
}

class NodePropertiesApi {
    constructor(client) {
        this.client = client;
    }
    /** List node custom properties */
    async list(nodeId, params) {
        const query = buildQueryString$2([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'visibility', value: params.visibility, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$2(appApiPath(`/drive/nodes/${serializePathParameter$2(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/properties`), query));
    }
    /** Create or update a node custom property */
    async set(nodeId, propertyKey, body) {
        return this.client.put(appApiPath(`/drive/nodes/${serializePathParameter$2(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/properties/${serializePathParameter$2(propertyKey, { name: 'propertyKey', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    /** Delete a node custom property */
    async delete(nodeId, propertyKey, params) {
        const query = buildQueryString$2([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'visibility', value: params.visibility, style: 'form', explode: true, allowReserved: false },
            { name: 'operatorId', value: params.operatorId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString$2(appApiPath(`/drive/nodes/${serializePathParameter$2(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/properties/${serializePathParameter$2(propertyKey, { name: 'propertyKey', style: 'simple', explode: false })}`), query));
    }
}
function createNodePropertiesApi(client) {
    return new NodePropertiesApi(client);
}
function appendQueryString$2(path, rawQueryString) {
    const query = rawQueryString.replace(/^\?+/, '');
    if (!query) {
        return path;
    }
    return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}
function serializePathParameter$2(value, spec) {
    if (value === undefined || value === null) {
        return '';
    }
    const style = spec.style || 'simple';
    if (Array.isArray(value)) {
        return serializePathArray$2(spec.name, value, style, spec.explode);
    }
    if (typeof value === 'object') {
        return serializePathObject$2(spec.name, value, style, spec.explode);
    }
    return pathPrefix$2(spec.name, style) + encodePathValue$2(serializePathPrimitive$2(value));
}
function serializePathArray$2(name, values, style, explode) {
    const serialized = values
        .filter((item) => item !== undefined && item !== null)
        .map((item) => encodePathValue$2(serializePathPrimitive$2(item)));
    if (serialized.length === 0) {
        return pathPrefix$2(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? serialized.map((item) => `;${name}=${item}`).join('')
            : `;${name}=${serialized.join(',')}`;
    }
    return pathPrefix$2(name, style) + serialized.join(explode ? '.' : ',');
}
function serializePathObject$2(name, value, style, explode) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return pathPrefix$2(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? entries.map(([key, entryValue]) => `;${encodePathValue$2(key)}=${encodePathValue$2(serializePathPrimitive$2(entryValue))}`).join('')
            : `;${name}=${entries.flatMap(([key, entryValue]) => [encodePathValue$2(key), encodePathValue$2(serializePathPrimitive$2(entryValue))]).join(',')}`;
    }
    const serialized = explode
        ? entries.map(([key, entryValue]) => `${encodePathValue$2(key)}=${encodePathValue$2(serializePathPrimitive$2(entryValue))}`).join(style === 'label' ? '.' : ',')
        : entries.flatMap(([key, entryValue]) => [encodePathValue$2(key), encodePathValue$2(serializePathPrimitive$2(entryValue))]).join(',');
    return pathPrefix$2(name, style) + serialized;
}
function pathPrefix$2(name, style, _objectValue) {
    if (style === 'label')
        return '.';
    if (style === 'matrix')
        return `;${name}`;
    return '';
}
function encodePathValue$2(value) {
    return encodeURIComponent(value);
}
function serializePathPrimitive$2(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function buildQueryString$2(parameters) {
    const pairs = [];
    for (const parameter of parameters) {
        appendSerializedParameter$2(pairs, parameter);
    }
    return pairs.join('&');
}
function appendSerializedParameter$2(pairs, parameter) {
    if (parameter.value === undefined || parameter.value === null) {
        return;
    }
    if (parameter.contentType) {
        pairs.push(`${encodeQueryComponent$2(parameter.name)}=${encodeQueryValue$2(JSON.stringify(parameter.value), parameter.allowReserved)}`);
        return;
    }
    const style = parameter.style || 'form';
    if (style === 'deepObject') {
        appendDeepObjectParameter$2(pairs, parameter.name, parameter.value, parameter.allowReserved);
        return;
    }
    if (Array.isArray(parameter.value)) {
        appendArrayParameter$2(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    if (typeof parameter.value === 'object') {
        appendObjectParameter$2(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    pairs.push(`${encodeQueryComponent$2(parameter.name)}=${encodeQueryValue$2(serializePrimitive$2(parameter.value), parameter.allowReserved)}`);
}
function appendArrayParameter$2(pairs, name, value, style, explode, allowReserved) {
    const values = value
        .filter((item) => item !== undefined && item !== null)
        .map((item) => serializePrimitive$2(item));
    if (values.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const item of values) {
            pairs.push(`${encodeQueryComponent$2(name)}=${encodeQueryValue$2(item, allowReserved)}`);
        }
        return;
    }
    pairs.push(`${encodeQueryComponent$2(name)}=${encodeQueryValue$2(values.join(','), allowReserved)}`);
}
function appendObjectParameter$2(pairs, name, value, style, explode, allowReserved) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const [key, entryValue] of entries) {
            pairs.push(`${encodeQueryComponent$2(key)}=${encodeQueryValue$2(serializePrimitive$2(entryValue), allowReserved)}`);
        }
        return;
    }
    const serialized = entries.flatMap(([key, entryValue]) => [key, serializePrimitive$2(entryValue)]).join(',');
    pairs.push(`${encodeQueryComponent$2(name)}=${encodeQueryValue$2(serialized, allowReserved)}`);
}
function appendDeepObjectParameter$2(pairs, name, value, allowReserved) {
    if (!value || typeof value !== 'object' || Array.isArray(value)) {
        pairs.push(`${encodeQueryComponent$2(name)}=${encodeQueryValue$2(serializePrimitive$2(value), allowReserved)}`);
        return;
    }
    for (const [key, entryValue] of Object.entries(value)) {
        if (entryValue === undefined || entryValue === null) {
            continue;
        }
        pairs.push(`${encodeQueryComponent$2(`${name}[${key}]`)}=${encodeQueryValue$2(serializePrimitive$2(entryValue), allowReserved)}`);
    }
}
function serializePrimitive$2(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function encodeQueryComponent$2(value) {
    return encodeURIComponent(value);
}
function encodeQueryValue$2(value, allowReserved) {
    const encoded = encodeURIComponent(value);
    if (!allowReserved) {
        return encoded;
    }
    return encoded.replace(/%3A/gi, ':')
        .replace(/%2F/gi, '/')
        .replace(/%3F/gi, '?')
        .replace(/%23/gi, '#')
        .replace(/%5B/gi, '[')
        .replace(/%5D/gi, ']')
        .replace(/%40/gi, '@')
        .replace(/%21/gi, '!')
        .replace(/%24/gi, '$')
        .replace(/%26/gi, '&')
        .replace(/%27/gi, "'")
        .replace(/%28/gi, '(')
        .replace(/%29/gi, ')')
        .replace(/%2A/gi, '*')
        .replace(/%2B/gi, '+')
        .replace(/%2C/gi, ',')
        .replace(/%3B/gi, ';')
        .replace(/%3D/gi, '=');
}

class NodesShortcutsApi {
    constructor(client) {
        this.client = client;
    }
    /** Create a shortcut node */
    async create(body) {
        return this.client.post(appApiPath(`/drive/nodes/shortcuts`), body, undefined, undefined, 'application/json');
    }
}
class NodesApi {
    constructor(client) {
        this.client = client;
        this.shortcuts = new NodesShortcutsApi(client);
    }
}
function createNodesApi(client) {
    return new NodesApi(client);
}

class WatchChannelsNodesApi {
    constructor(client) {
        this.client = client;
    }
    /** Create a push notification channel for a Drive node */
    async watch(nodeId, body) {
        return this.client.post(appApiPath(`/drive/nodes/${serializePathParameter$1(nodeId, { name: 'nodeId', style: 'simple', explode: false })}/watch`), body, undefined, undefined, 'application/json');
    }
}
class WatchChannelsChangesApi {
    constructor(client) {
        this.client = client;
    }
    /** Create a push notification channel for Drive changes */
    async watch(body) {
        return this.client.post(appApiPath(`/drive/changes/watch`), body, undefined, undefined, 'application/json');
    }
}
class WatchChannelsApi {
    constructor(client) {
        this.client = client;
        this.changes = new WatchChannelsChangesApi(client);
        this.nodes = new WatchChannelsNodesApi(client);
    }
    /** List Drive watch channels */
    async list(params) {
        const query = buildQueryString$1([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'resourceType', value: params.resourceType, style: 'form', explode: true, allowReserved: false },
            { name: 'lifecycleStatus', value: params.lifecycleStatus, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'pageToken', value: params.pageToken, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$1(appApiPath(`/drive/watch_channels`), query));
    }
    /** Get a Drive watch channel */
    async get(channelId, params) {
        const query = buildQueryString$1([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString$1(appApiPath(`/drive/watch_channels/${serializePathParameter$1(channelId, { name: 'channelId', style: 'simple', explode: false })}`), query));
    }
    /** Stop a Drive watch channel */
    async stop(channelId, body) {
        return this.client.post(appApiPath(`/drive/watch_channels/${serializePathParameter$1(channelId, { name: 'channelId', style: 'simple', explode: false })}/stop`), body, undefined, undefined, 'application/json');
    }
}
function createWatchChannelsApi(client) {
    return new WatchChannelsApi(client);
}
function appendQueryString$1(path, rawQueryString) {
    const query = rawQueryString.replace(/^\?+/, '');
    if (!query) {
        return path;
    }
    return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}
function serializePathParameter$1(value, spec) {
    if (value === undefined || value === null) {
        return '';
    }
    const style = spec.style || 'simple';
    if (Array.isArray(value)) {
        return serializePathArray$1(spec.name, value, style, spec.explode);
    }
    if (typeof value === 'object') {
        return serializePathObject$1(spec.name, value, style, spec.explode);
    }
    return pathPrefix$1(spec.name, style) + encodePathValue$1(serializePathPrimitive$1(value));
}
function serializePathArray$1(name, values, style, explode) {
    const serialized = values
        .filter((item) => item !== undefined && item !== null)
        .map((item) => encodePathValue$1(serializePathPrimitive$1(item)));
    if (serialized.length === 0) {
        return pathPrefix$1(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? serialized.map((item) => `;${name}=${item}`).join('')
            : `;${name}=${serialized.join(',')}`;
    }
    return pathPrefix$1(name, style) + serialized.join(explode ? '.' : ',');
}
function serializePathObject$1(name, value, style, explode) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return pathPrefix$1(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? entries.map(([key, entryValue]) => `;${encodePathValue$1(key)}=${encodePathValue$1(serializePathPrimitive$1(entryValue))}`).join('')
            : `;${name}=${entries.flatMap(([key, entryValue]) => [encodePathValue$1(key), encodePathValue$1(serializePathPrimitive$1(entryValue))]).join(',')}`;
    }
    const serialized = explode
        ? entries.map(([key, entryValue]) => `${encodePathValue$1(key)}=${encodePathValue$1(serializePathPrimitive$1(entryValue))}`).join(style === 'label' ? '.' : ',')
        : entries.flatMap(([key, entryValue]) => [encodePathValue$1(key), encodePathValue$1(serializePathPrimitive$1(entryValue))]).join(',');
    return pathPrefix$1(name, style) + serialized;
}
function pathPrefix$1(name, style, _objectValue) {
    if (style === 'label')
        return '.';
    if (style === 'matrix')
        return `;${name}`;
    return '';
}
function encodePathValue$1(value) {
    return encodeURIComponent(value);
}
function serializePathPrimitive$1(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function buildQueryString$1(parameters) {
    const pairs = [];
    for (const parameter of parameters) {
        appendSerializedParameter$1(pairs, parameter);
    }
    return pairs.join('&');
}
function appendSerializedParameter$1(pairs, parameter) {
    if (parameter.value === undefined || parameter.value === null) {
        return;
    }
    if (parameter.contentType) {
        pairs.push(`${encodeQueryComponent$1(parameter.name)}=${encodeQueryValue$1(JSON.stringify(parameter.value), parameter.allowReserved)}`);
        return;
    }
    const style = parameter.style || 'form';
    if (style === 'deepObject') {
        appendDeepObjectParameter$1(pairs, parameter.name, parameter.value, parameter.allowReserved);
        return;
    }
    if (Array.isArray(parameter.value)) {
        appendArrayParameter$1(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    if (typeof parameter.value === 'object') {
        appendObjectParameter$1(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    pairs.push(`${encodeQueryComponent$1(parameter.name)}=${encodeQueryValue$1(serializePrimitive$1(parameter.value), parameter.allowReserved)}`);
}
function appendArrayParameter$1(pairs, name, value, style, explode, allowReserved) {
    const values = value
        .filter((item) => item !== undefined && item !== null)
        .map((item) => serializePrimitive$1(item));
    if (values.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const item of values) {
            pairs.push(`${encodeQueryComponent$1(name)}=${encodeQueryValue$1(item, allowReserved)}`);
        }
        return;
    }
    pairs.push(`${encodeQueryComponent$1(name)}=${encodeQueryValue$1(values.join(','), allowReserved)}`);
}
function appendObjectParameter$1(pairs, name, value, style, explode, allowReserved) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const [key, entryValue] of entries) {
            pairs.push(`${encodeQueryComponent$1(key)}=${encodeQueryValue$1(serializePrimitive$1(entryValue), allowReserved)}`);
        }
        return;
    }
    const serialized = entries.flatMap(([key, entryValue]) => [key, serializePrimitive$1(entryValue)]).join(',');
    pairs.push(`${encodeQueryComponent$1(name)}=${encodeQueryValue$1(serialized, allowReserved)}`);
}
function appendDeepObjectParameter$1(pairs, name, value, allowReserved) {
    if (!value || typeof value !== 'object' || Array.isArray(value)) {
        pairs.push(`${encodeQueryComponent$1(name)}=${encodeQueryValue$1(serializePrimitive$1(value), allowReserved)}`);
        return;
    }
    for (const [key, entryValue] of Object.entries(value)) {
        if (entryValue === undefined || entryValue === null) {
            continue;
        }
        pairs.push(`${encodeQueryComponent$1(`${name}[${key}]`)}=${encodeQueryValue$1(serializePrimitive$1(entryValue), allowReserved)}`);
    }
}
function serializePrimitive$1(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function encodeQueryComponent$1(value) {
    return encodeURIComponent(value);
}
function encodeQueryValue$1(value, allowReserved) {
    const encoded = encodeURIComponent(value);
    if (!allowReserved) {
        return encoded;
    }
    return encoded.replace(/%3A/gi, ':')
        .replace(/%2F/gi, '/')
        .replace(/%3F/gi, '?')
        .replace(/%23/gi, '#')
        .replace(/%5B/gi, '[')
        .replace(/%5D/gi, ']')
        .replace(/%40/gi, '@')
        .replace(/%21/gi, '!')
        .replace(/%24/gi, '$')
        .replace(/%26/gi, '&')
        .replace(/%27/gi, "'")
        .replace(/%28/gi, '(')
        .replace(/%29/gi, ')')
        .replace(/%2A/gi, '*')
        .replace(/%2B/gi, '+')
        .replace(/%2C/gi, ',')
        .replace(/%3B/gi, ';')
        .replace(/%3D/gi, '=');
}

class AssetsAssetRelationsApi {
    constructor(client) {
        this.client = client;
    }
    /** Create an asset relation */
    async create(assetId, body) {
        return this.client.post(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/relations`), body, undefined, undefined, 'application/json');
    }
    /** Delete an asset relation */
    async delete(assetId, relationId, params) {
        const query = buildQueryString([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/relations/${serializePathParameter(relationId, { name: 'relationId', style: 'simple', explode: false })}`), query));
    }
}
class AssetsAssetCollectionItemsApi {
    constructor(client) {
        this.client = client;
    }
    /** Add an asset to a collection */
    async create(collectionId, body) {
        return this.client.post(appApiPath(`/assets/collections/${serializePathParameter(collectionId, { name: 'collectionId', style: 'simple', explode: false })}/items`), body, undefined, undefined, 'application/json');
    }
    /** Remove an asset from a collection */
    async delete(collectionId, itemId, params) {
        const query = buildQueryString([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.delete(appendQueryString(appApiPath(`/assets/collections/${serializePathParameter(collectionId, { name: 'collectionId', style: 'simple', explode: false })}/items/${serializePathParameter(itemId, { name: 'itemId', style: 'simple', explode: false })}`), query));
    }
}
class AssetsAssetCollectionsApi {
    constructor(client) {
        this.client = client;
    }
    /** List asset collections */
    async list(params) {
        const query = buildQueryString([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'cursor', value: params.cursor, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString(appApiPath(`/assets/collections`), query));
    }
    /** Create an asset collection */
    async create(body) {
        return this.client.post(appApiPath(`/assets/collections`), body, undefined, undefined, 'application/json');
    }
}
class AssetsApi {
    constructor(client) {
        this.client = client;
        this.assetCollections = new AssetsAssetCollectionsApi(client);
        this.assetCollectionItems = new AssetsAssetCollectionItemsApi(client);
        this.assetRelations = new AssetsAssetRelationsApi(client);
    }
    /** List global assets */
    async list(params) {
        const query = buildQueryString([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
            { name: 'cursor', value: params.cursor, style: 'form', explode: true, allowReserved: false },
            { name: 'pageSize', value: params.pageSize, style: 'form', explode: true, allowReserved: false },
            { name: 'kind', value: params.kind, style: 'form', explode: true, allowReserved: false },
            { name: 'sourceType', value: params.sourceType, style: 'form', explode: true, allowReserved: false },
            { name: 'q', value: params.q, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString(appApiPath(`/assets`), query));
    }
    /** Create a global asset metadata record */
    async create(body) {
        return this.client.post(appApiPath(`/assets`), body, undefined, undefined, 'application/json');
    }
    /** Get a global asset */
    async get(assetId, params) {
        const query = buildQueryString([
            { name: 'tenantId', value: params.tenantId, style: 'form', explode: true, allowReserved: false },
        ]);
        return this.client.get(appendQueryString(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}`), query));
    }
    /** Update a global asset */
    async update(assetId, body) {
        return this.client.patch(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}`), body, undefined, undefined, 'application/json');
    }
    /** Archive a global asset */
    async archive(assetId, body) {
        return this.client.post(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/archive`), body, undefined, undefined, 'application/json');
    }
    /** Restore an archived global asset */
    async restore(assetId, body) {
        return this.client.post(appApiPath(`/assets/${serializePathParameter(assetId, { name: 'assetId', style: 'simple', explode: false })}/restore`), body, undefined, undefined, 'application/json');
    }
}
function createAssetsApi(client) {
    return new AssetsApi(client);
}
function appendQueryString(path, rawQueryString) {
    const query = rawQueryString.replace(/^\?+/, '');
    if (!query) {
        return path;
    }
    return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}
function serializePathParameter(value, spec) {
    if (value === undefined || value === null) {
        return '';
    }
    const style = spec.style || 'simple';
    if (Array.isArray(value)) {
        return serializePathArray(spec.name, value, style, spec.explode);
    }
    if (typeof value === 'object') {
        return serializePathObject(spec.name, value, style, spec.explode);
    }
    return pathPrefix(spec.name, style) + encodePathValue(serializePathPrimitive(value));
}
function serializePathArray(name, values, style, explode) {
    const serialized = values
        .filter((item) => item !== undefined && item !== null)
        .map((item) => encodePathValue(serializePathPrimitive(item)));
    if (serialized.length === 0) {
        return pathPrefix(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? serialized.map((item) => `;${name}=${item}`).join('')
            : `;${name}=${serialized.join(',')}`;
    }
    return pathPrefix(name, style) + serialized.join(explode ? '.' : ',');
}
function serializePathObject(name, value, style, explode) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return pathPrefix(name, style);
    }
    if (style === 'matrix') {
        return explode
            ? entries.map(([key, entryValue]) => `;${encodePathValue(key)}=${encodePathValue(serializePathPrimitive(entryValue))}`).join('')
            : `;${name}=${entries.flatMap(([key, entryValue]) => [encodePathValue(key), encodePathValue(serializePathPrimitive(entryValue))]).join(',')}`;
    }
    const serialized = explode
        ? entries.map(([key, entryValue]) => `${encodePathValue(key)}=${encodePathValue(serializePathPrimitive(entryValue))}`).join(style === 'label' ? '.' : ',')
        : entries.flatMap(([key, entryValue]) => [encodePathValue(key), encodePathValue(serializePathPrimitive(entryValue))]).join(',');
    return pathPrefix(name, style) + serialized;
}
function pathPrefix(name, style, _objectValue) {
    if (style === 'label')
        return '.';
    if (style === 'matrix')
        return `;${name}`;
    return '';
}
function encodePathValue(value) {
    return encodeURIComponent(value);
}
function serializePathPrimitive(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function buildQueryString(parameters) {
    const pairs = [];
    for (const parameter of parameters) {
        appendSerializedParameter(pairs, parameter);
    }
    return pairs.join('&');
}
function appendSerializedParameter(pairs, parameter) {
    if (parameter.value === undefined || parameter.value === null) {
        return;
    }
    if (parameter.contentType) {
        pairs.push(`${encodeQueryComponent(parameter.name)}=${encodeQueryValue(JSON.stringify(parameter.value), parameter.allowReserved)}`);
        return;
    }
    const style = parameter.style || 'form';
    if (style === 'deepObject') {
        appendDeepObjectParameter(pairs, parameter.name, parameter.value, parameter.allowReserved);
        return;
    }
    if (Array.isArray(parameter.value)) {
        appendArrayParameter(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    if (typeof parameter.value === 'object') {
        appendObjectParameter(pairs, parameter.name, parameter.value, style, parameter.explode, parameter.allowReserved);
        return;
    }
    pairs.push(`${encodeQueryComponent(parameter.name)}=${encodeQueryValue(serializePrimitive(parameter.value), parameter.allowReserved)}`);
}
function appendArrayParameter(pairs, name, value, style, explode, allowReserved) {
    const values = value
        .filter((item) => item !== undefined && item !== null)
        .map((item) => serializePrimitive(item));
    if (values.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const item of values) {
            pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(item, allowReserved)}`);
        }
        return;
    }
    pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(values.join(','), allowReserved)}`);
}
function appendObjectParameter(pairs, name, value, style, explode, allowReserved) {
    const entries = Object.entries(value).filter(([, entryValue]) => entryValue !== undefined && entryValue !== null);
    if (entries.length === 0) {
        return;
    }
    if (style === 'form' && explode) {
        for (const [key, entryValue] of entries) {
            pairs.push(`${encodeQueryComponent(key)}=${encodeQueryValue(serializePrimitive(entryValue), allowReserved)}`);
        }
        return;
    }
    const serialized = entries.flatMap(([key, entryValue]) => [key, serializePrimitive(entryValue)]).join(',');
    pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(serialized, allowReserved)}`);
}
function appendDeepObjectParameter(pairs, name, value, allowReserved) {
    if (!value || typeof value !== 'object' || Array.isArray(value)) {
        pairs.push(`${encodeQueryComponent(name)}=${encodeQueryValue(serializePrimitive(value), allowReserved)}`);
        return;
    }
    for (const [key, entryValue] of Object.entries(value)) {
        if (entryValue === undefined || entryValue === null) {
            continue;
        }
        pairs.push(`${encodeQueryComponent(`${name}[${key}]`)}=${encodeQueryValue(serializePrimitive(entryValue), allowReserved)}`);
    }
}
function serializePrimitive(value) {
    if (value instanceof Date) {
        return value.toISOString();
    }
    if (typeof value === 'object') {
        return JSON.stringify(value);
    }
    return String(value);
}
function encodeQueryComponent(value) {
    return encodeURIComponent(value);
}
function encodeQueryValue(value, allowReserved) {
    const encoded = encodeURIComponent(value);
    if (!allowReserved) {
        return encoded;
    }
    return encoded.replace(/%3A/gi, ':')
        .replace(/%2F/gi, '/')
        .replace(/%3F/gi, '?')
        .replace(/%23/gi, '#')
        .replace(/%5B/gi, '[')
        .replace(/%5D/gi, ']')
        .replace(/%40/gi, '@')
        .replace(/%21/gi, '!')
        .replace(/%24/gi, '$')
        .replace(/%26/gi, '&')
        .replace(/%27/gi, "'")
        .replace(/%28/gi, '(')
        .replace(/%29/gi, ')')
        .replace(/%2A/gi, '*')
        .replace(/%2B/gi, '+')
        .replace(/%2C/gi, ',')
        .replace(/%3B/gi, ';')
        .replace(/%3D/gi, '=');
}

class SdkworkAppClient {
    constructor(config) {
        this.httpClient = createHttpClient(config);
        this.drive = createDriveApi(this.httpClient);
        this.nodeLabels = createNodeLabelsApi(this.httpClient);
        this.nodeProperties = createNodePropertiesApi(this.httpClient);
        this.nodes = createNodesApi(this.httpClient);
        this.watchChannels = createWatchChannelsApi(this.httpClient);
        this.assets = createAssetsApi(this.httpClient);
    }
    setApiKey(apiKey) {
        this.httpClient.setApiKey(apiKey);
        return this;
    }
    setAuthToken(token) {
        this.httpClient.setAuthToken(token);
        return this;
    }
    setAccessToken(token) {
        this.httpClient.setAccessToken(token);
        return this;
    }
    setTokenManager(manager) {
        this.httpClient.setTokenManager(manager);
        return this;
    }
    get http() {
        return this.httpClient;
    }
}
function createClient(config) {
    return new SdkworkAppClient(config);
}

class BaseApi {
    constructor(http, basePath) {
        this.http = http;
        this.basePath = basePath;
    }
    async get(path, params, headers) {
        return this.http.get(`${this.basePath}${path}`, params, headers);
    }
    async post(path, body, params, headers, contentType) {
        return this.http.post(`${this.basePath}${path}`, body, params, headers, contentType);
    }
    async put(path, body, params, headers, contentType) {
        return this.http.put(`${this.basePath}${path}`, body, params, headers, contentType);
    }
    async delete(path, params, headers) {
        return this.http.delete(`${this.basePath}${path}`, params, headers);
    }
    async patch(path, body, params, headers, contentType) {
        return this.http.patch(`${this.basePath}${path}`, body, params, headers, contentType);
    }
    async request(method, path, body, params, headers, contentType) {
        return this.http.request(`${this.basePath}${path}`, { method: method, body, params, headers, contentType });
    }
}

Object.defineProperty(exports, "DEFAULT_TIMEOUT", {
    enumerable: true,
    get: function () { return sdkCommon.DEFAULT_TIMEOUT; }
});
Object.defineProperty(exports, "DefaultAuthTokenManager", {
    enumerable: true,
    get: function () { return sdkCommon.DefaultAuthTokenManager; }
});
Object.defineProperty(exports, "SUCCESS_CODES", {
    enumerable: true,
    get: function () { return sdkCommon.SUCCESS_CODES; }
});
Object.defineProperty(exports, "createTokenManager", {
    enumerable: true,
    get: function () { return sdkCommon.createTokenManager; }
});
exports.AssetsApi = AssetsApi;
exports.BaseApi = BaseApi;
exports.DriveApi = DriveApi;
exports.HttpClient = HttpClient;
exports.NodeLabelsApi = NodeLabelsApi;
exports.NodePropertiesApi = NodePropertiesApi;
exports.NodesApi = NodesApi;
exports.SdkworkAppClient = SdkworkAppClient;
exports.WatchChannelsApi = WatchChannelsApi;
exports.appApiPath = appApiPath;
exports.createAssetsApi = createAssetsApi;
exports.createClient = createClient;
exports.createDriveApi = createDriveApi;
exports.createHttpClient = createHttpClient;
exports.createNodeLabelsApi = createNodeLabelsApi;
exports.createNodePropertiesApi = createNodePropertiesApi;
exports.createNodesApi = createNodesApi;
exports.createWatchChannelsApi = createWatchChannelsApi;
