// @sdkwork/appbase-app-sdk/src/api/auth.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';

import type { AppbaseApiResult, AppbaseOperationCommand } from '../types';


export class AuthSessionsCurrentApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }


/** Sessions current delete. */
  async delete(): Promise<AppbaseApiResult> {
    return this.client.delete<AppbaseApiResult>(appApiPath(`/auth/sessions/current`));
  }

/** Sessions current retrieve. */
  async retrieve(): Promise<AppbaseApiResult> {
    return this.client.get<AppbaseApiResult>(appApiPath(`/auth/sessions/current`));
  }

/** Sessions current update. */
  async update(body?: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.patch<AppbaseApiResult>(appApiPath(`/auth/sessions/current`), body, undefined, undefined, 'application/json');
  }
}

export class AuthSessionsApi {
  private client: HttpClient;
  public readonly current: AuthSessionsCurrentApi;
  
  constructor(client: HttpClient) { 
    this.client = client;
    this.current = new AuthSessionsCurrentApi(client); 
  }


/** Sessions create. */
  async create(body: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.post<AppbaseApiResult>(appApiPath(`/auth/sessions`), body, undefined, undefined, 'application/json');
  }

/** Sessions refresh. */
  async refresh(body: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.post<AppbaseApiResult>(appApiPath(`/auth/sessions/refresh`), body, undefined, undefined, 'application/json');
  }
}

export class AuthRegistrationsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }


/** Registrations create. */
  async create(body: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.post<AppbaseApiResult>(appApiPath(`/auth/registrations`), body, undefined, undefined, 'application/json');
  }
}

export class AuthPasswordResetsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }


/** Password Resets create. */
  async create(body: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.post<AppbaseApiResult>(appApiPath(`/auth/password_resets`), body, undefined, undefined, 'application/json');
  }
}

export class AuthPasswordResetRequestsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }


/** Password Reset Requests create. */
  async create(body: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.post<AppbaseApiResult>(appApiPath(`/auth/password_reset_requests`), body, undefined, undefined, 'application/json');
  }
}

export class AuthOauthSessionsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }


/** Oauth Sessions create. */
  async create(body: AppbaseOperationCommand): Promise<AppbaseApiResult> {
    return this.client.post<AppbaseApiResult>(appApiPath(`/auth/oauth_sessions`), body, undefined, undefined, 'application/json');
  }
}

export class AuthOauthAuthorizationUrlsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }


/** Oauth Authorization Urls retrieve. */
  async retrieve(): Promise<AppbaseApiResult> {
    return this.client.get<AppbaseApiResult>(appApiPath(`/auth/oauth_authorization_urls`));
  }
}

export class AuthApi {
  private client: HttpClient;
  public readonly oauthAuthorizationUrls: AuthOauthAuthorizationUrlsApi;
  public readonly oauthSessions: AuthOauthSessionsApi;
  public readonly passwordResetRequests: AuthPasswordResetRequestsApi;
  public readonly passwordResets: AuthPasswordResetsApi;
  public readonly registrations: AuthRegistrationsApi;
  public readonly sessions: AuthSessionsApi;
  
  constructor(client: HttpClient) { 
    this.client = client;
    this.oauthAuthorizationUrls = new AuthOauthAuthorizationUrlsApi(client);
    this.oauthSessions = new AuthOauthSessionsApi(client);
    this.passwordResetRequests = new AuthPasswordResetRequestsApi(client);
    this.passwordResets = new AuthPasswordResetsApi(client);
    this.registrations = new AuthRegistrationsApi(client);
    this.sessions = new AuthSessionsApi(client);
  }

}

export function createAuthApi(client: HttpClient): AuthApi {
  return new AuthApi(client);
}

function appendQueryString(path: string, rawQueryString: string): string {
  const query = rawQueryString.replace(/^\?+/, '');
  if (!query) {
    return path;
  }
  return path.includes('?') ? `${path}&${query}` : `${path}?${query}`;
}
