// @sdkwork/appbase-app-sdk/src/sdk.ts
import { HttpClient, createHttpClient } from './http/client';
import type { SdkworkAppConfig } from './types/common';
import type { AuthTokenManager } from '@sdkwork/sdk-common';

import { AuthApi, createAuthApi } from './api/auth';
import { IamApi, createIamApi } from './api/iam';
import { OpenPlatformApi, createOpenPlatformApi } from './api/open-platform';
import { SystemApi, createSystemApi } from './api/system';

export class SdkworkAppClient {
  private httpClient: HttpClient;

  public readonly auth: AuthApi;
  public readonly iam: IamApi;
  public readonly openPlatform: OpenPlatformApi;
  public readonly system: SystemApi;

  constructor(config: SdkworkAppConfig) {
    this.httpClient = createHttpClient(config);
    this.auth = createAuthApi(this.httpClient);

    this.iam = createIamApi(this.httpClient);

    this.openPlatform = createOpenPlatformApi(this.httpClient);

    this.system = createSystemApi(this.httpClient);
  }

  setApiKey(apiKey: string): this {
    this.httpClient.setApiKey(apiKey);
    return this;
  }

  setAuthToken(token: string): this {
    this.httpClient.setAuthToken(token);
    return this;
  }

  setAccessToken(token: string): this {
    this.httpClient.setAccessToken(token);
    return this;
  }

  setTokenManager(manager: AuthTokenManager): this {
    this.httpClient.setTokenManager(manager);
    return this;
  }

  get http(): HttpClient {
    return this.httpClient;
  }
}

export function createClient(config: SdkworkAppConfig): SdkworkAppClient {
  return new SdkworkAppClient(config);
}

export default SdkworkAppClient;
