// @sdkwork/core-pc-react/dist/index.js
import { useMemo as e, useSyncExternalStore as t } from "react";
//#region src/internal/helpers.ts
var n = 3e4;
function r(...e) {
	for (let t of e) {
		let e = typeof t == "string" ? t.trim() : "";
		if (e) return e;
	}
}
function i(e) {
	return (e || "").trim();
}
function a(e) {
	let t = i(e);
	return t ? t.replace(/^Bearer\s+/i, "").trim() : "";
}
function o(e) {
	return i(e).replace(/\/+$/g, "");
}
function s(e, t = n) {
	let r = Number(e);
	return !Number.isFinite(r) || r <= 0 ? t : r;
}
function c(e) {
	let t = Number(e);
	return !Number.isFinite(t) || t <= 0 ? null : t;
}
function l(e, t) {
	if (typeof e == "boolean") return e;
	let n = i(typeof e == "string" ? e : String(t)).toLowerCase();
	return n === "true" || n === "1";
}
function u(e) {
	return JSON.parse(JSON.stringify(e));
}
function d(e) {
	if (!e) return null;
	try {
		return JSON.parse(e);
	} catch {
		return null;
	}
}
function f(e) {
	return JSON.stringify(e);
}
function ee(e) {
	switch (e) {
		case "production": return "https://api.sdkwork.com";
		case "test": return "https://api-test.sdkwork.com";
		case "staging": return "https://staging-api.sdkwork.com";
		default: return "https://api-dev.sdkwork.com";
	}
}
function te(e) {
	let t = o(e);
	if (!t) return "";
	try {
		let e = new URL(t);
		return e.protocol = e.protocol === "https:" ? "wss:" : "ws:", e.pathname = "/ws", e.search = "", e.hash = "", e.toString().replace(/\/+$/g, "");
	} catch {
		return "";
	}
}
function ne(...e) {
	let t = r(...e)?.toLowerCase();
	return t === "production" || t === "prod" ? "production" : t === "test" ? "test" : t === "staging" || t === "stage" ? "staging" : "development";
}
function re(e, t, n, r) {
	return r || (i(e) && !i(t) && !i(n) ? "apikey" : "dual-token");
}
//#endregion
//#region src/env/index.ts
var p = /* @__PURE__ */ "VITE_APP_ENV.VITE_APP_OWNER_MODE.VITE_OWNER_MODE.VITE_API_BASE_URL.VITE_APP_BASE_URL.VITE_APP_ROOT_API_BASE_URL.VITE_APP_TENANT_API_BASE_URL.VITE_APP_ORGANIZATION_API_BASE_URL.VITE_ROOT_API_BASE_URL.VITE_APP_ROOT_API_BASE_URL.VITE_TENANT_API_BASE_URL.VITE_APP_TENANT_API_BASE_URL.VITE_ORGANIZATION_API_BASE_URL.VITE_APP_ORGANIZATION_API_BASE_URL.VITE_IM_WS_URL.VITE_APP_IM_WS_URL.VITE_ACCESS_TOKEN.VITE_APP_ACCESS_TOKEN.VITE_APP_ROOT_ACCESS_TOKEN.VITE_APP_TENANT_ACCESS_TOKEN.VITE_APP_ORGANIZATION_ACCESS_TOKEN.VITE_ROOT_ACCESS_TOKEN.VITE_APP_ROOT_ACCESS_TOKEN.VITE_TENANT_ACCESS_TOKEN.VITE_APP_TENANT_ACCESS_TOKEN.VITE_ORGANIZATION_ACCESS_TOKEN.VITE_APP_ORGANIZATION_ACCESS_TOKEN.VITE_API_KEY.VITE_TIMEOUT.VITE_TENANT_ID.VITE_APP_TENANT_ID.VITE_ORGANIZATION_ID.VITE_APP_ORGANIZATION_ID.VITE_PLATFORM.VITE_APP_PLATFORM.VITE_APP_NAME.VITE_APP_VERSION.VITE_DISTRIBUTION_ID.VITE_APP_ID.VITE_RELEASE_CHANNEL.VITE_ENABLE_STARTUP_UPDATE_CHECK.VITE_DEBUG.VITE_LOG_LEVEL".split(".");
function m() {
	return {
		BASE_URL: "/",
		DEV: !1,
		MODE: "production",
		PROD: !0,
		SSR: !1
	};
}
function h() {
	return globalThis.process?.env ?? {};
}
function g() {
	return {
		...h(),
		...m()
	};
}
function _(e = []) {
	let t = globalThis;
	return e.reduce((e, n) => {
		let r = t[n];
		return !r || typeof r != "object" ? e : {
			...e,
			...r
		};
	}, {});
}
function ie() {
	if (typeof window > "u") return !1;
	let e = window;
	return !!(e.__TAURI__ || e.__TAURI_IPC__ || e.__TAURI_INTERNALS__);
}
function ae(e, t) {
	return t ? "desktop" : r(i(e.VITE_PLATFORM), i(e.VITE_APP_PLATFORM), i(e.SDKWORK_PLATFORM)) || "web";
}
function oe(e) {
	let t = r(i(e.VITE_APP_OWNER_MODE), i(e.VITE_OWNER_MODE), i(e.SDKWORK_OWNER_MODE))?.toLowerCase();
	return t === "organization" || t === "org" ? "organization" : t === "tenant" ? "tenant" : t === "root" ? "root" : r(i(e.VITE_TENANT_ID), i(e.VITE_APP_TENANT_ID), i(e.SDKWORK_TENANT_ID), i(e.VITE_TENANT_ACCESS_TOKEN), i(e.VITE_APP_TENANT_ACCESS_TOKEN), i(e.VITE_TENANT_API_BASE_URL), i(e.VITE_APP_TENANT_API_BASE_URL)) ? "tenant" : r(i(e.VITE_ORGANIZATION_ID), i(e.VITE_APP_ORGANIZATION_ID), i(e.SDKWORK_ORGANIZATION_ID), i(e.VITE_ORGANIZATION_ACCESS_TOKEN), i(e.VITE_APP_ORGANIZATION_ACCESS_TOKEN), i(e.VITE_ORGANIZATION_API_BASE_URL), i(e.VITE_APP_ORGANIZATION_API_BASE_URL)) ? "organization" : "root";
}
function se(e, t) {
	let n = o(r(i(e.VITE_API_BASE_URL), i(e.VITE_APP_API_BASE_URL), i(e.VITE_APP_BASE_URL), i(e.SDKWORK_API_BASE_URL), ee(t)));
	return {
		default: n,
		root: o(r(i(e.VITE_ROOT_API_BASE_URL), i(e.VITE_APP_ROOT_API_BASE_URL), n)),
		tenant: o(r(i(e.VITE_TENANT_API_BASE_URL), i(e.VITE_APP_TENANT_API_BASE_URL), n)),
		organization: o(r(i(e.VITE_ORGANIZATION_API_BASE_URL), i(e.VITE_APP_ORGANIZATION_API_BASE_URL), n))
	};
}
function ce(e) {
	let t = a(r(i(e.VITE_ACCESS_TOKEN), i(e.VITE_APP_ACCESS_TOKEN)));
	return {
		default: t,
		root: a(r(i(e.VITE_ROOT_ACCESS_TOKEN), i(e.VITE_APP_ROOT_ACCESS_TOKEN), t)),
		tenant: a(r(i(e.VITE_TENANT_ACCESS_TOKEN), i(e.VITE_APP_TENANT_ACCESS_TOKEN), t)),
		organization: a(r(i(e.VITE_ORGANIZATION_ACCESS_TOKEN), i(e.VITE_APP_ORGANIZATION_ACCESS_TOKEN), t))
	};
}
function le(e, t) {
	return e === "tenant" ? t.tenant ?? t.default : e === "organization" ? t.organization ?? t.default : t.root ?? t.default;
}
function v(e = g()) {
	let t = ne(i(e.VITE_APP_ENV), i(e.MODE), i(e.NODE_ENV)), n = oe(e), u = se(e, t), d = ce(e), f = o(le(n, u) || ee(t)), p = a(le(n, d)), m = i(r(i(e.VITE_API_KEY), i(e.SDKWORK_API_KEY))), h = ie(), _ = ae(e, h), v = o(r(i(e.VITE_IM_WS_URL), i(e.VITE_APP_IM_WS_URL), te(f))), y = i(r(i(e.VITE_TENANT_ID), i(e.VITE_APP_TENANT_ID), i(e.SDKWORK_TENANT_ID))), b = i(r(i(e.VITE_ORGANIZATION_ID), i(e.VITE_APP_ORGANIZATION_ID), i(e.SDKWORK_ORGANIZATION_ID)));
	return {
		appEnv: t,
		mode: t,
		isDev: t === "development",
		isTest: t === "test",
		isStaging: t === "staging",
		isProduction: t === "production",
		metadata: {
			name: i(e.VITE_APP_NAME) || "SDKWork Desktop",
			version: i(e.VITE_APP_VERSION) || "0.1.0"
		},
		log: {
			debug: l(e.VITE_DEBUG, t === "development"),
			level: i(e.VITE_LOG_LEVEL) || (t === "development" ? "debug" : t === "test" ? "info" : "warn")
		},
		owner: {
			mode: n,
			tenantId: y,
			organizationId: b
		},
		api: {
			baseUrl: f,
			baseUrls: u,
			timeout: s(r(i(e.VITE_TIMEOUT), i(e.SDKWORK_TIMEOUT)))
		},
		auth: {
			apiKey: m,
			accessToken: p,
			accessTokens: d,
			mode: re(m, p)
		},
		realtime: { imWsUrl: v },
		update: {
			appId: c(i(e.VITE_APP_ID)),
			releaseChannel: i(e.VITE_RELEASE_CHANNEL) || "stable",
			enableStartupCheck: l(e.VITE_ENABLE_STARTUP_UPDATE_CHECK, !0)
		},
		distribution: { id: i(e.VITE_DISTRIBUTION_ID) === "cn" ? "cn" : "global" },
		platform: {
			id: _,
			isDesktop: h || _ === "desktop",
			isTauri: h
		},
		vite: {
			isDev: l(e.DEV, t === "development"),
			isProd: l(e.PROD, t === "production")
		}
	};
}
//#endregion
//#region src/internal/runtimeState.ts
var y = "sdkwork.core.pc-react.auth-token", b = "core.pc-react.access-token", x = "sdkwork.core.pc-react.refresh-token", S = "sdkwork.core.pc-react.im-session", ue = "sdkwork_token", C = "sdkwork_refresh_token", w = {
	authToken: [ue],
	accessToken: [],
	refreshToken: [C],
	runtimeSession: [
		"sdkwork-notes-auth-session",
		"sdkwork-drive-auth-session",
		"claw-studio-auth-session"
	],
	imSession: []
}, T = /* @__PURE__ */ new Map(), E = {}, de = null, fe = null, D = null, pe = null, O = null, me = 0, he = null, ge = /* @__PURE__ */ new Set(), _e = /* @__PURE__ */ new Set();
function ve() {
	if (typeof window > "u") return null;
	try {
		return window.localStorage;
	} catch {
		return null;
	}
}
function ye() {
	return {
		getItem(e) {
			return T.get(e) ?? null;
		},
		setItem(e, t) {
			T.set(e, t);
		},
		removeItem(e) {
			T.delete(e);
		}
	};
}
function k(e = [], t = []) {
	return Array.from(new Set([...e, ...t].filter((e) => e && e.trim())));
}
function be(e, t) {
	if (!(!e && !t)) return {
		authToken: k(e?.authToken, t?.authToken),
		accessToken: k(e?.accessToken, t?.accessToken),
		refreshToken: k(e?.refreshToken, t?.refreshToken),
		runtimeSession: k(e?.runtimeSession, t?.runtimeSession),
		imSession: k(e?.imSession, t?.imSession)
	};
}
function xe(e, t) {
	if (!(!e && !t)) return {
		defaults: {
			...e?.defaults ?? {},
			...t?.defaults ?? {}
		},
		storageKey: t?.storageKey ?? e?.storageKey
	};
}
function A() {
	return {
		authToken: k(w.authToken, E.legacyStorageKeys?.authToken),
		accessToken: k(w.accessToken, E.legacyStorageKeys?.accessToken),
		refreshToken: k(w.refreshToken, E.legacyStorageKeys?.refreshToken),
		runtimeSession: k(w.runtimeSession, E.legacyStorageKeys?.runtimeSession),
		imSession: k(w.imSession, E.legacyStorageKeys?.imSession)
	};
}
function Se(e, t) {
	return {
		envSource: t.envSource ? {
			...e.envSource ?? {},
			...t.envSource
		} : e.envSource,
		envGlobalKeys: t.envGlobalKeys ? k(e.envGlobalKeys, t.envGlobalKeys) : e.envGlobalKeys,
		storage: t.storage ?? e.storage,
		legacyStorageKeys: be(e.legacyStorageKeys, t.legacyStorageKeys),
		preferences: xe(e.preferences, t.preferences),
		appClient: t.appClient ?? e.appClient,
		appClientFactory: t.appClientFactory ?? e.appClientFactory,
		appClientCompatAliases: {
			...e.appClientCompatAliases ?? {},
			...t.appClientCompatAliases ?? {}
		},
		headersResolver: t.headersResolver ?? e.headersResolver,
		appConfigOverrides: {
			...e.appConfigOverrides ?? {},
			...t.appConfigOverrides ?? {}
		},
		imConfigOverrides: {
			...e.imConfigOverrides ?? {},
			...t.imConfigOverrides ?? {}
		}
	};
}
function j() {
	me += 1;
	for (let e of ge) e();
}
function Ce(e) {
	for (let t of _e) t(e);
	j();
}
function we() {
	return {
		...g(),
		..._(E.envGlobalKeys ?? []),
		...E.envSource ?? {}
	};
}
function M(e) {
	try {
		return F().getItem(e) ?? void 0;
	} catch {
		return;
	}
}
function N(e) {
	for (let t of e) {
		let e = M(t);
		if (e !== void 0) return e;
	}
}
function Te() {
	for (let e of A().runtimeSession) {
		let t = d(M(e));
		if (!t) continue;
		let n = a(t.authToken), r = a(t.accessToken), i = typeof t.refreshToken == "string" ? t.refreshToken.trim() : "";
		if (!(!n && !r && !i)) return {
			authToken: n || void 0,
			accessToken: r || void 0,
			refreshToken: i || void 0
		};
	}
	return {};
}
function P(e, t) {
	try {
		if (t && t.trim()) {
			F().setItem(e, t.trim());
			return;
		}
		F().removeItem(e);
	} catch {}
}
function Ee() {
	for (let e of A().accessToken) P(e, void 0);
}
function F() {
	return E.storage ? E.storage : ve() || ye();
}
function I() {
	return E;
}
function De(e = {}) {
	return E = Se(E, e), Fe(), j(), E;
}
function L(e) {
	return ge.add(e), () => {
		ge.delete(e);
	};
}
function R() {
	return me;
}
function z() {
	return de ||= v(we()), de;
}
function Oe(e, t = B()) {
	let n = {}, r = a(t.authToken), i = a(t.accessToken);
	r && (n.Authorization = `Bearer ${r}`), i && (n["Access-Token"] = i);
	let o = E.headersResolver?.({
		env: z(),
		session: t,
		target: e
	});
	return o ? Object.entries(o).reduce((e, [t, n]) => {
		let r = typeof n == "string" ? n.trim() : "";
		return !t.trim() || !r || (e[t] = r), e;
	}, n) : n;
}
function B() {
	let e = z(), t = Te(), n = a(M(y) || t.authToken || N(A().authToken)), r = a(M(b) || t.accessToken || N(A().accessToken)), i = a(D?.accessToken || pe?.accessToken), o = (M(x) || t.refreshToken || N(A().refreshToken) || "").trim(), s = d(M(S));
	return !O && s && (O = s), {
		authToken: n || void 0,
		accessToken: r || i || e.auth.accessToken || void 0,
		refreshToken: o || void 0,
		im: O ? u(O) : void 0
	};
}
function ke(e) {
	let t = B(), n = z(), r = e.authToken === void 0 ? a(t.authToken) : a(e.authToken), i = e.accessToken === void 0 ? a(t.accessToken || n.auth.accessToken) : a(e.accessToken), o = e.refreshToken === void 0 ? (t.refreshToken || "").trim() : (e.refreshToken || "").trim();
	return P(y, r || void 0), e.accessToken === void 0 ? P(b, a(M(b)) || void 0) : !i || i === n.auth.accessToken ? (P(b, void 0), Ee()) : (P(b, i), Ee()), P(x, o || void 0), e.im && je(e.im), j(), B();
}
function Ae() {
	P(y, void 0), P(b, void 0), P(x, void 0), P(S, void 0);
	for (let e of [
		...A().authToken,
		...A().accessToken,
		...A().refreshToken,
		...A().runtimeSession,
		...A().imSession
	]) P(e, void 0);
	O = null, j();
}
function je(e) {
	O = e ? u(e) : null, e ? P(S, f(e)) : P(S, void 0), j();
}
function Me(e, t) {
	fe = e, D = t;
}
function Ne() {
	return fe ?? null;
}
function Pe() {
	return D ?? null;
}
function Fe() {
	de = null, fe = null, D = null, pe = null, O = null, he?.(), he = null, Ce("idle");
}
function Ie(e = {}) {
	let { clearStorage: t = !0, clearConfiguration: n = !0 } = e;
	t && (T.clear(), Ae()), n && (E = {}), Fe(), j();
}
//#endregion
//#region src/app/index.ts
var Le = {
	analytics: "analytic",
	assets: "asset",
	coupons: "coupon",
	notes: "note",
	orders: "order",
	payments: "payment",
	projects: "project",
	settings: "setting",
	workspaces: "workspace"
}, Re = /* @__PURE__ */ new WeakMap();
function ze(e = {}) {
	return {
		...I().appConfigOverrides ?? {},
		...e
	};
}
function V(e, t, n) {
	e.setAuthToken(a(t.authToken)), e.setAccessToken(a(t.accessToken || n));
}
function Be(e, t) {
	let n = B(), r = ze(e);
	return {
		...n,
		authToken: r.authToken === void 0 ? a(n.authToken) : a(r.authToken),
		accessToken: r.accessToken === void 0 ? a(n.accessToken || t) : a(r.accessToken)
	};
}
function Ve() {
	return { ...I().appClientCompatAliases ?? {} };
}
function He(e, t = Ve()) {
	let n = Object.entries(t).filter(([e, t]) => e.trim() && t.trim() && e !== t).sort(([e], [t]) => e.localeCompare(t));
	if (n.length === 0) return e;
	let r = JSON.stringify(n);
	if (Re.get(e) === r) return e;
	for (let [t, r] of n) Object.defineProperty(e, t, {
		configurable: !0,
		enumerable: !1,
		get() {
			return e[r];
		}
	});
	return Re.set(e, r), e;
}
function Ue(e = {}, t = { includeRuntimeSession: !0 }) {
	let n = z(), r = t.includeRuntimeSession ? B() : void 0, i = ze(e), o = a(i.accessToken || r?.accessToken || n.auth.accessToken), s = (i.apiKey || n.auth.apiKey || "").trim(), c = a(i.authToken || r?.authToken);
	return {
		env: n.appEnv,
		ownerMode: n.owner.mode,
		baseUrl: i.baseUrl || n.api.baseUrl,
		timeout: i.timeout ?? n.api.timeout,
		apiKey: s || void 0,
		accessToken: o || void 0,
		authToken: c || void 0,
		tenantId: (i.tenantId ?? n.owner.tenantId) || void 0,
		organizationId: (i.organizationId ?? n.owner.organizationId) || void 0,
		platform: i.platform ?? n.platform.id,
		tokenManager: i.tokenManager,
		authMode: re(s, o, c, i.authMode),
		headers: {
			...Oe("app", {
				...r ?? {},
				authToken: c || void 0,
				accessToken: o || void 0
			}),
			...i.headers ?? {}
		}
	};
}
function We(e, t = {}) {
	let n = a(t.accessToken || e.auth.accessToken), r = (t.apiKey || e.auth.apiKey || "").trim(), i = a(t.authToken);
	return {
		env: e.appEnv,
		ownerMode: e.owner.mode,
		baseUrl: t.baseUrl || e.api.baseUrl,
		timeout: t.timeout ?? e.api.timeout,
		apiKey: r || void 0,
		accessToken: n || void 0,
		authToken: i || void 0,
		tenantId: (t.tenantId ?? e.owner.tenantId) || void 0,
		organizationId: (t.organizationId ?? e.owner.organizationId) || void 0,
		platform: t.platform ?? e.platform.id,
		tokenManager: t.tokenManager,
		authMode: re(r, n, i, t.authMode),
		headers: { ...t.headers ?? {} }
	};
}
function Ge(e, t = {}) {
	return We(v(e), t);
}
function Ke(e) {
	return a(v(e).auth.accessToken);
}
function qe(e = {}) {
	return Ue(e, { includeRuntimeSession: !0 });
}
function H(e = {}) {
	return Ue(e, { includeRuntimeSession: !1 });
}
function Je(e) {
	let t = I(), n = t.appClientFactory?.(e) ?? t.appClient;
	if (!n) throw Error("[SDKWork Core] appClientFactory is required. Configure the runtime with a product app SDK client factory or an approved composed app client.");
	return He(n);
}
function Ye(e = {}) {
	let t = H(e), n = Je(t);
	return V(n, Be(e, t.accessToken || ""), t.accessToken || ""), Me(n, t), n;
}
function Xe() {
	return Ne() || Ye();
}
function U() {
	return Pe();
}
function Ze() {
	return a(B().accessToken) || a(U()?.accessToken) || a(z().auth.accessToken);
}
function Qe(e = {}) {
	let t = H(e), n = Je(t);
	return V(n, Be(e, t.accessToken || ""), t.accessToken || ""), n;
}
function $e(e = {}) {
	if (Object.keys(e).length > 0) return Qe(e);
	let t = Xe(), n = U() || H();
	return V(t, B(), n.accessToken || ""), t;
}
function et(e = B()) {
	let t = Ne(), n = U();
	!t || !n || V(t, e, n.accessToken || "");
}
function tt(e) {
	let t = Xe();
	t.setAuthToken(a(e.authToken)), t.setAccessToken(a(e.accessToken || Ze()));
}
//#endregion
//#region src/internal/preferencesState.ts
var nt = "sdkwork.core.pc-react.preferences", rt = !0, it = 0, at = !1, ot = !1, st = "", ct = "", W = xt(), lt = /* @__PURE__ */ new Set();
function ut(e) {
	let t = typeof e == "string" ? e.trim().toLowerCase() : "";
	if (t === "dark" || t === "light" || t === "system") return t;
}
function dt(e) {
	let t = typeof e == "string" ? e.trim().toLowerCase() : "";
	if (t === "green-tech" || t === "lobster" || t === "rose" || t === "tech-blue" || t === "violet" || t === "zinc") return t;
}
function G(e) {
	return (typeof e == "string" ? e.trim() : "") || void 0;
}
function ft(e) {
	let t = typeof e == "string" ? e.trim() : "";
	if (t) return t.toLowerCase() === "system" ? "system" : G(t);
}
function pt() {
	if (!(typeof navigator > "u")) return G(navigator.language);
}
function mt() {
	if (!(typeof document > "u")) return G(document.documentElement.lang);
}
function ht(e = {}) {
	return G(e.documentLanguage) || G(e.navigatorLanguage) || mt() || pt() || "en-US";
}
function gt() {
	return I().preferences;
}
function _t() {
	return G(gt()?.storageKey) || nt;
}
function vt() {
	return d(F().getItem(_t()));
}
function yt(e = {}) {
	let t = gt()?.defaults ?? {}, n = ft(t.localePreference ?? t.locale) || "system";
	return {
		locale: n === "system" ? ht(e) : n,
		localePreference: n,
		themeColor: dt(t.themeColor) || "lobster",
		themeSelection: ut(t.themeSelection) || "system"
	};
}
function bt(e, t = {}) {
	let n = yt(t), r = ft(e?.localePreference ?? e?.locale) || n.localePreference;
	return {
		locale: r === "system" ? ht(t) : r,
		localePreference: r,
		themeColor: dt(e?.themeColor) || n.themeColor,
		themeSelection: ut(e?.themeSelection) || n.themeSelection
	};
}
function xt() {
	return typeof window > "u" || typeof window.matchMedia != "function" ? rt : window.matchMedia("(prefers-color-scheme: dark)").matches;
}
function St(e) {
	return f(e);
}
function Ct(e) {
	return f(e);
}
function K(e = {}) {
	return bt(vt(), e);
}
function wt(e = {}) {
	let t = K(e), n = e.prefersDark ?? W, r = t.themeSelection === "system" ? n ? "dark" : "light" : t.themeSelection;
	return {
		...t,
		colorMode: r
	};
}
function Tt() {
	it += 1;
	for (let e of lt) e();
}
function Et() {
	st = St(K()), ct = Ct(wt());
}
function q() {
	let e = St(K()), t = Ct(wt());
	e === st && t === ct || (st = e, ct = t, Tt());
}
function J() {
	at ||= (Et(), L(() => {
		q();
	}), !0);
}
function Dt() {
	if (!ot) {
		if (W = xt(), typeof window < "u" && typeof window.matchMedia == "function") {
			let e = window.matchMedia("(prefers-color-scheme: dark)"), t = (e) => {
				let t = e.matches;
				W !== t && (W = t, q());
			};
			e.addEventListener?.("change", t), e.addListener?.(t);
		}
		ot = !0;
	}
}
function Ot() {
	return it;
}
function kt() {
	return J(), K();
}
function At(e) {
	J();
	let t = bt({
		...kt(),
		...e,
		...e.localePreference === void 0 ? e.locale === void 0 ? {} : { localePreference: e.locale } : { localePreference: e.localePreference }
	});
	return F().setItem(_t(), f(t)), q(), t;
}
function jt() {
	J(), F().removeItem(_t()), q();
}
function Y(e = {}) {
	return wt(e);
}
function Mt(e) {
	return J(), Dt(), lt.add(e), () => {
		lt.delete(e);
	};
}
//#endregion
//#region src/runtime/shell-bridge.ts
function Nt(e) {
	let t = String(e).trim().toLowerCase();
	return t.startsWith("ar") || t.startsWith("fa") || t.startsWith("he") || t.startsWith("ps") || t.startsWith("ur");
}
function Pt(e) {
	return String(e || "").trim() || "en-US";
}
function Ft(e) {
	if (e instanceof Date) return Number.isNaN(e.getTime()) ? null : e;
	if (typeof e == "number") {
		let t = new Date(e);
		return Number.isNaN(t.getTime()) ? null : t;
	}
	if (typeof e == "string") {
		let t = e.trim();
		if (!t) return null;
		if (/^\d{4}-\d{2}-\d{2}$/.test(t)) {
			let [e, n, r] = t.split("-").map(Number), i = new Date(e, n - 1, r);
			return Number.isNaN(i.getTime()) ? null : i;
		}
		let n = new Date(t);
		return Number.isNaN(n.getTime()) ? null : n;
	}
	return null;
}
function X(e) {
	return At(e), Y();
}
function It(e) {
	return Nt(Pt(e)) ? "rtl" : "ltr";
}
function Lt(e) {
	let t = Pt(e);
	return {
		formatDate(e, n) {
			let r = Ft(e);
			return r ? new Intl.DateTimeFormat(t, n).format(r) : "";
		},
		formatDateTime(e, n) {
			let r = Ft(e);
			return r ? new Intl.DateTimeFormat(t, n ?? {
				dateStyle: "medium",
				timeStyle: "short"
			}).format(r) : "";
		},
		locale: t
	};
}
function Rt() {
	let e = Y(), t = e.locale;
	return {
		actions: {
			patchPreferences: X,
			setLocalePreference(e) {
				return X({ localePreference: e });
			},
			setThemeColor(e) {
				return X({ themeColor: e });
			},
			setThemeSelection(e) {
				return X({ themeSelection: e });
			}
		},
		dir: It(t),
		env: z(),
		formatters: Lt(t),
		locale: t,
		preferences: e,
		session: B()
	};
}
//#endregion
//#region src/runtime/index.ts
function zt(e = {}) {
	return De(e);
}
function Bt(e = {}) {
	Ie(e);
}
function Vt(e) {
	let t = ke(e);
	return et(t), t;
}
var Ht = Vt;
function Z() {
	return B();
}
var Ut = Z;
function Wt() {
	Ae(), et(B());
}
var Gt = {
	authToken: ue,
	refreshToken: C
};
//#endregion
//#region src/hooks/index.ts
function Q() {
	t(L, R, R);
}
function $() {
	t(Mt, Ot, Ot);
}
function Kt() {
	return Q(), $e();
}
function qt() {
	return Q(), z();
}
function Jt() {
	return Q(), Z();
}
function Yt() {
	return $(), kt();
}
function Xt() {
	return $(), Y();
}
function Zt() {
	return Q(), $(), e(() => Rt(), [
		z(),
		Y(),
		Z()
	]);
}
//#endregion
export { Le as SDKWORK_PC_REACT_DEFAULT_APP_CLIENT_COMPAT_ALIASES, p as SDKWORK_PC_REACT_ENV_KEYS, ue as SDKWORK_PC_REACT_LEGACY_AUTH_TOKEN_STORAGE_KEY, C as SDKWORK_PC_REACT_LEGACY_REFRESH_TOKEN_STORAGE_KEY, Gt as SDKWORK_PC_REACT_LEGACY_STORAGE_KEYS, tt as applyAppClientSessionTokens, et as applyRuntimeSessionToAppClient, Wt as clearPcReactRuntimeSession, jt as clearPcReactShellPreferences, zt as configurePcReactRuntime, qe as createAppClientConfig, Ge as createAppClientConfigFromEnv, v as createPcReactEnvConfig, Lt as createPcReactLocaleFormatting, Qe as createScopedAppClient, He as decorateAppClientCompatAliases, Xe as getAppClient, U as getAppClientConfig, $e as getAppClientWithSession, z as getPcReactEnv, R as getPcReactRuntimeVersion, Ye as initAppClient, Ht as persistPcReactRuntimeSession, At as persistPcReactShellPreferences, Vt as persistRuntimeSession, g as readPcReactEnvSource, _ as readPcReactNamedGlobalEnvSources, Ut as readPcReactRuntimeSession, Rt as readPcReactShellBridgeValue, kt as readPcReactShellPreferences, Z as readRuntimeSession, Bt as resetPcReactRuntime, Ze as resolveAppClientAccessToken, Ke as resolveAppClientAccessTokenFromEnv, It as resolvePcReactLocaleDirection, Y as resolvePcReactShellPreferences, L as subscribePcReactRuntime, Mt as subscribePcReactShellPreferences, Kt as useAppClient, qt as usePcReactEnv, Xt as usePcReactResolvedShellPreferences, Jt as usePcReactRuntimeSession, Zt as usePcReactShellBridgeValue, Yt as usePcReactShellPreferences };
